{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 396,
   "id": "f7e12fdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(hms)\n",
    "library(tidyverse)\n",
    "library(readxl)\n",
    "library(ggplot2)\n",
    "library(SIAMCAT)\n",
    "library(data.table)\n",
    "library(vegan)\n",
    "library(RColorBrewer)\n",
    "source('packages.r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 397,
   "id": "21470477-f299-4e57-9443-390a68ca541b",
   "metadata": {},
   "outputs": [],
   "source": [
    "rarefyDepth <- 500"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 398,
   "id": "3f3f1622",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Ananthakrishnan_2017_CellHostMicrobe\"\n",
      "[1] \"Backhed_2015_CellHostMicrobe\"\n",
      "[1] \"Bahram_2018_Nature\"\n",
      "[1] \"Baxter_2016_GenomeMed\"\n",
      "[1] \"Bedarf_2017_GenomeMed\"\n",
      "[1] \"Brito_2016_Nature\"\n",
      "[1] \"Bullman_2017_Science\"\n",
      "[1] \"Burns_2017_GenomeMed\"\n",
      "[1] \"Callahan_2017_PNAS\"\n",
      "[1] \"Castellarin_2012_GenomeMed\"\n",
      "[1] \"CastroNaller_2015_PeerJ\"\n",
      "[1] \"Chen_2018_ISME\"\n",
      "[1] \"Clausen_2018_JAMADermatology\"\n",
      "[1] \"Daniel_metadata\"\n",
      "[1] \"Dzunkova_2019_NatMicrobiol\"\n",
      "[1] \"Fan_2018_Gut\"\n",
      "[1] \"Feng_2015_NatComm\"\n",
      "[1] \"Fettweis_2019_NatMed\"\n",
      "[1] \"Flemer_2018_Gut\"\n",
      "[1] \"Forsell_2017_BMCMicrobiol\"\n",
      "[1] \"Frankel_2017_Neoplasia\"\n",
      "[1] \"Franzosa_NatMed_2019\"\n",
      "[1] \"Goodrich_2014_Cell\"\n",
      "[1] \"Hale_2018_GenomeMed\"\n",
      "[1] \"Hall_2017_GenomeMedicine\"\n",
      "[1] \"Hannigan_2018_mBio\"\n",
      "[1] \"Hayes_2018_JAMAOncology\"\n",
      "[1] \"He_2017_GigaScience\"\n",
      "[1] \"Heintz-Buschart_2016_NatMicrobiol\"\n",
      "[1] \"HMP_2012_Nature_stool\"\n",
      "[1] \"Hoyles_2018_NatMed\"\n",
      "[1] \"Hsu_2016_mSystems\"\n",
      "[1] \"Jie_2017_NatComm\"\n",
      "[1] \"Johnson_2019_CellHostMicrobe\"\n",
      "[1] \"Karlsson_2013_Nature\"\n",
      "[1] \"Kostic_2012_GenomeRes\"\n",
      "[1] \"Kostic_2015_CellHostMicrobe\"\n",
      "[1] \"Kushugolova_2018_BMJOpen\"\n",
      "[1] \"Lappan_2018_BMCBiology\"\n",
      "[1] \"LeChatelier_2013_Nature\"\n",
      "[1] \"Lewis_2015_CellHostMicrobe\"\n",
      "[1] \"Li_2017_Microbiome\"\n",
      "[1] \"Liu_2016_SciRep\"\n",
      "[1] \"Liu_2022_NatMicrobiol\"\n",
      "[1] \"Lloyd-Price_2019_Nature\"\n",
      "[1] \"Loomba_2017_CellMetab\"\n",
      "[1] \"Mars_2020_Cell\"\n",
      "[1] \"Matson_2018_Science\"\n",
      "[1] \"meta_cn_t2d.tsv\"\n",
      "[1] \"meta_hmp2.tsv\"\n",
      "[1] \"Milieu_interieur\"\n",
      "[1] \"Nakatsu_2015_NatComm\"\n",
      "[1] \"Nielsen_2014_NatBiotechnol\"\n",
      "[1] \"Obregon-Tito_2015_NatComm\"\n",
      "[1] \"OKeefe_2015_NatComm\"\n",
      "[1] \"Palleja_2019_NatMicrobiol\"\n",
      "[1] \"Pasolli_2019_Cell\"\n",
      "[1] \"Poyet_2019_NatMedicine\"\n",
      "[1] \"Purcell_2017_SciRep\"\n",
      "[1] \"Qin_2012_Nature\"\n",
      "[1] \"Qin_2012_Nature_2\"\n",
      "[1] \"Qin_2014_Nature\"\n",
      "[1] \"Rampelli_2015_CurrBiol\"\n",
      "[1] \"Raymond_2016_ISMEJ\"\n",
      "[1] \"Routy_2018_Science\"\n",
      "[1] \"Schirmer_2016_Cell\"\n",
      "[1] \"Suez_2018_Cell\"\n",
      "[1] \"Thomas_2019_NatMedicine\"\n",
      "[1] \"Urbaniak_2016_ApplEnvironMicrobiol\"\n",
      "[1] \"Vatanen_2016_Cell\"\n",
      "[1] \"Vincnet_2016_Microbiome\"\n",
      "[1] \"Vogtmann_2016_PLoS\"\n",
      "[1] \"Voigt_2015_GenomeBiol\"\n",
      "[1] \"Vujkovic-Cvijin_2010_Nature\"\n",
      "[1] \"Wang_2016_EurJGastroenterolHepatol\"\n",
      "[1] \"Warren_2013_Microbiome\"\n",
      "[1] \"Wen_2017_GenomeBiol\"\n",
      "[1] \"Wirbel_2019_NatMedicine\"\n",
      "[1] \"Xie_2016_CellSyst\"\n",
      "[1] \"Xie_2016_CellSyst_2\"\n",
      "[1] \"Yachida_2019_NatMedicine\"\n",
      "[1] \"Ye_2018_Microbiome\"\n",
      "[1] \"Yu_2017_Gut\"\n",
      "[1] \"Zeevi_2015_Cell\"\n",
      "[1] \"Zeller_2014_MSB\"\n",
      "[1] \"Zhang_205_NatMed\"\n",
      "[1] \"Zhou_2019_Nature\"\n",
      "[1] \"Zhu_2018_Microbiome\"\n",
      "[1] \"Zhu_2020_NatComm\"\n",
      "[1] \"Zmora_2018_Cell\"\n"
     ]
    }
   ],
   "source": [
    "for (f in list.files(\"/g/scb2/zeller/SHARED/DATA/metadata/\")){\n",
    "    print(f)\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "638678af",
   "metadata": {},
   "source": [
    "### TODO\n",
    "\n",
    "- Link TwinsUK [done]\n",
    "- Match sampleIDs between profiles and metadata for Bedarf...\n",
    "- Include US-CRC (Vogtmann)?\n",
    "- Include Forslund cohort? Understand which samples in that cohort are actually new"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 399,
   "id": "8d509dcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_map <- c(\"Backhed_2015_CellHostMicrobe\" = \"meta_backhed.tsv\",\n",
    "                    \"Bedarf_2017_GenomeMed\" = \"meta_bedarf.tsv\",\n",
    "                    \"Feng_2015_NatComm\" = \"cleaned_metadata.tsv\",\n",
    "                    \"Franzosa_NatMed_2019\" = \"meta_franzosa.tsv\",\n",
    "                    \"He_2017_GigaScience\" = \"meta_he.tsv\",\n",
    "                    \"Hoyles_2018_NatMed\" = \"meta_hoyles.tsv\",\n",
    "                    \"Jie_2017_NatComm\" = \"meta_jie_minimal.tsv\",\n",
    "                    \"Karlsson_2013_Nature\" = \"meta_SE-T2D.tsv\",\n",
    "                    \"Kushugolova_2018_BMJOpen\" = \"meta_kushugolova.tsv\",\n",
    "                    \"LeChatelier_2013_Nature\" = \"meta_lechatelier.tsv\",\n",
    "                    \"Lewis_2015_CellHostMicrobe\" = \"meta_lewis.tsv\",\n",
    "                    \"Li_2017_Microbiome\" = \"meta_Li.tsv\",\n",
    "                    \"Lloyd-Price_2019_Nature\" = \"meta_lloyd_price_metaG.tsv\",\n",
    "                    \"Loomba_2017_CellMetab\" = \"meta_loomba.tsv\",\n",
    "                    \"Palleja_2019_NatMicrobiol\" = \"meta_palleja.tsv\",\n",
    "                    \"Poyet_2019_NatMedicine\" = \"meta_poyet.tsv\",\n",
    "                    \"Qin_2012_Nature_2\" = \"meta_cn_t2d.tsv\",\n",
    "                    \"Qin_2014_Nature\" = \"meta_qin.tsv\",\n",
    "                    \"Schirmer_2016_Cell\" = \"meta_schirmer.tsv\",\n",
    "                    \"Suez_2018_Cell\" = \"meta_suez_raw.tsv\",\n",
    "                    \"Thomas_2019_NatMedicine\" = \"cleaned_metadata.tsv\",\n",
    "                    \"Vincnet_2016_Microbiome\" = \"meta_vincent.tsv\",\n",
    "                    \"Wen_2017_GenomeBiol\" = \"meta_wen.tsv\",\n",
    "                    \"Wirbel_2019_NatMedicine\" = \"cleaned_metadata.tsv\",\n",
    "                    \"Yachida_2019_NatMedicine\" = \"meta_Yachida_2019.tsv\",\n",
    "                    \"Yu_2017_Gut\" = \"cleaned_metadata.tsv\",\n",
    "                    \"Zeevi_2015_Cell\" = \"meta_zeevi_updated.tsv\",\n",
    "                    \"Zeller_2014_MSB\" = \"cleaned_metadata.tsv\",\n",
    "                    \"Zhang_205_NatMed\" = \"meta_zhang.tsv\",\n",
    "                    \"Zhu_2018_Microbiome\" = \"meta_zhu.tsv\",\n",
    "                    \"Nielsen_2014_NatBiotechnol\" = \"meta_metaHIT-IBD.tsv\",\n",
    "                    \"Xie_2016_CellSyst_2\" = \"meta_xie.tsv\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 400,
   "id": "9dac86e9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Profile basename:Backhed_2015_CellHostMicrobe\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Description = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Pair = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Bedarf_2017_GenomeMed\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  AB_taken = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Height = \u001b[32mcol_double()\u001b[39m,\n",
      "  Weight = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Smoking = \u001b[32mcol_double()\u001b[39m,\n",
      "  Diet = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Feng_2015_NatComm\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[32mcol_double()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  AJCC_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  TNM_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  Localization = \u001b[31mcol_character()\u001b[39m,\n",
      "  Study = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_rel_to_colonoscopy = \u001b[31mcol_character()\u001b[39m,\n",
      "  FOBT = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Diabetes = \u001b[32mcol_double()\u001b[39m,\n",
      "  Vegetarian = \u001b[32mcol_double()\u001b[39m,\n",
      "  Smoking = \u001b[32mcol_double()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Franzosa_NatMed_2019\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Fecal.calprotein = \u001b[32mcol_double()\u001b[39m,\n",
      "  Antibiotics = \u001b[31mcol_character()\u001b[39m,\n",
      "  Mesalamine = \u001b[31mcol_character()\u001b[39m,\n",
      "  Steroids = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Immunosuppressant = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:He_2017_GigaScience\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[31mcol_character()\u001b[39m,\n",
      "  Height = \u001b[31mcol_character()\u001b[39m,\n",
      "  Weight = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Nutritional_status = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Hoyles_2018_NatMed\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Sex = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Jie_2017_NatComm\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Smoking = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Karlsson_2013_Nature\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Fasting_glucose = \u001b[31mcol_character()\u001b[39m,\n",
      "  Adiponectin = \u001b[31mcol_character()\u001b[39m,\n",
      "  Leptin = \u001b[31mcol_character()\u001b[39m,\n",
      "  `GLP-1` = \u001b[31mcol_character()\u001b[39m,\n",
      "  `FGF-19` = \u001b[31mcol_character()\u001b[39m,\n",
      "  `IL-1` = \u001b[31mcol_character()\u001b[39m,\n",
      "  CD163 = \u001b[31mcol_character()\u001b[39m,\n",
      "  Statins = \u001b[31mcol_character()\u001b[39m,\n",
      "  Insulin = \u001b[31mcol_character()\u001b[39m,\n",
      "  Medication = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Kushugolova_2018_BMJOpen\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Intervention = \u001b[31mcol_character()\u001b[39m,\n",
      "  Western = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:LeChatelier_2013_Nature\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Lewis_2015_CellHostMicrobe\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Antibiotics_visit = \u001b[31mcol_character()\u001b[39m,\n",
      "  Response = \u001b[31mcol_character()\u001b[39m,\n",
      "  Steroids = \u001b[31mcol_character()\u001b[39m,\n",
      "  Treatment = \u001b[31mcol_character()\u001b[39m,\n",
      "  Treatment_Specific = \u001b[31mcol_character()\u001b[39m,\n",
      "  Time = \u001b[32mcol_double()\u001b[39m,\n",
      "  Subject = \u001b[32mcol_double()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Li_2017_Microbiome\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Lloyd-Price_2019_Nature\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Loomba_2017_CellMetab\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Nielsen_2014_NatBiotechnol\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_day = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Palleja_2019_NatMicrobiol\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  Library_size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Poyet_2019_NatMedicine\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_size = \u001b[32mcol_double()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Qin_2012_Nature_2\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Qin_2014_Nature\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  HBV = \u001b[31mcol_character()\u001b[39m,\n",
      "  Alcohol = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Schirmer_2016_Cell\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Suez_2018_Cell\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Alias = \u001b[31mcol_character()\u001b[39m,\n",
      "  Description = \u001b[31mcol_character()\u001b[39m,\n",
      "  LibrarySelection = \u001b[31mcol_character()\u001b[39m,\n",
      "  sex = \u001b[31mcol_character()\u001b[39m,\n",
      "  tissue_type = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Participant = \u001b[32mcol_double()\u001b[39m,\n",
      "  Day = \u001b[32mcol_double()\u001b[39m,\n",
      "  phase = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Thomas_2019_NatMedicine\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  AJCC_stage = \u001b[33mcol_logical()\u001b[39m,\n",
      "  TNM_stage = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Localization = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Study = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_rel_to_colonoscopy = \u001b[33mcol_logical()\u001b[39m,\n",
      "  FOBT = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Diabetes = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Vegetarian = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Smoking = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Vincnet_2016_Microbiome\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Wen_2017_GenomeBio\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Disease_duration = \u001b[32mcol_double()\u001b[39m,\n",
      "  Antibiotics = \u001b[31mcol_character()\u001b[39m,\n",
      "  Peripheral_AS = \u001b[31mcol_character()\u001b[39m,\n",
      "  NSAIDs = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Wirbel_2019_NatMedicine\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  AJCC_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  TNM_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  Localization = \u001b[31mcol_character()\u001b[39m,\n",
      "  Study = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_rel_to_colonoscopy = \u001b[31mcol_character()\u001b[39m,\n",
      "  FOBT = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Diabetes = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Vegetarian = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Smoking = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Xie_2016_CellSyst_2\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  BGI_ID = \u001b[32mcol_double()\u001b[39m,\n",
      "  Twin_pair_number = \u001b[31mcol_character()\u001b[39m,\n",
      "  Zygosity = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sex = \u001b[33mcol_logical()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Age_of_separation = \u001b[32mcol_double()\u001b[39m,\n",
      "  Delivery_mode = \u001b[31mcol_character()\u001b[39m,\n",
      "  Menopause = \u001b[31mcol_character()\u001b[39m,\n",
      "  Vegetarian_diet = \u001b[31mcol_character()\u001b[39m,\n",
      "  Smoking = \u001b[31mcol_character()\u001b[39m,\n",
      "  Alcohol = \u001b[31mcol_character()\u001b[39m,\n",
      "  Outdoor_sports = \u001b[31mcol_character()\u001b[39m,\n",
      "  Metformin = \u001b[31mcol_character()\u001b[39m,\n",
      "  Medication = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Yachida_2019_NatMedicine\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Status = \u001b[31mcol_character()\u001b[39m,\n",
      "  Subject_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Brinkman Index` = \u001b[31mcol_character()\u001b[39m,\n",
      "  Alcohol = \u001b[31mcol_character()\u001b[39m,\n",
      "  `Tumor location` = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group_detailled = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Yu_2017_Gut\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  AJCC_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  TNM_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  Localization = \u001b[31mcol_character()\u001b[39m,\n",
      "  Study = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_rel_to_colonoscopy = \u001b[31mcol_character()\u001b[39m,\n",
      "  FOBT = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Diabetes = \u001b[32mcol_double()\u001b[39m,\n",
      "  Vegetarian = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Smoking = \u001b[32mcol_double()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Zeevi_2015_Cell\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[32mcol_double()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Zeller_2014_MSB\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m,\n",
      "  Country = \u001b[31mcol_character()\u001b[39m,\n",
      "  AJCC_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  TNM_stage = \u001b[31mcol_character()\u001b[39m,\n",
      "  Localization = \u001b[31mcol_character()\u001b[39m,\n",
      "  Study = \u001b[31mcol_character()\u001b[39m,\n",
      "  Sampling_rel_to_colonoscopy = \u001b[31mcol_character()\u001b[39m,\n",
      "  FOBT = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Diabetes = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Vegetarian = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Smoking = \u001b[33mcol_logical()\u001b[39m,\n",
      "  Library_Size = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Zhang_205_NatMed\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  External_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Individual_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Timepoint = \u001b[32mcol_double()\u001b[39m,\n",
      "  Gender = \u001b[31mcol_character()\u001b[39m,\n",
      "  Age = \u001b[32mcol_double()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m,\n",
      "  Treatment = \u001b[31mcol_character()\u001b[39m,\n",
      "  BMI = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"Profile basename:Zhu_2018_Microbiome\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  Sample_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Library_size = \u001b[32mcol_double()\u001b[39m,\n",
      "  Menopause = \u001b[31mcol_character()\u001b[39m,\n",
      "  Group = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "Warning message:\n",
      "“Missing column names filled in: 'X1' [1]”\n",
      "Parsed with column specification:\n",
      "cols(\n",
      "  .default = col_double(),\n",
      "  X1 = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n",
      "See spec(...) for full column specifications.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n",
      "[1] \"################################################\"\n"
     ]
    }
   ],
   "source": [
    "data <- list()\n",
    "for (file in list.files(\"/g/scb2/zeller/karcher/dysb_classif/data/siamcat/profiles\")){\n",
    "    base = str_split(file, \"[.]motus\")[[1]][1]\n",
    "    print(str_c(\"Profile basename:\", base))\n",
    "    me <- list.files(\"/g/scb2/zeller/SHARED/DATA/metadata/\")\n",
    "    me <- me[str_detect(me, base)]\n",
    "    if (length(me) == 0){\n",
    "        print(str_c(\"Metadata: \", \"### Unclear ###\"))\n",
    "    } else {\n",
    "        for (m in me){\n",
    "            metadata <- read_tsv(str_c(\"/g/scb2/zeller/SHARED/DATA/metadata/\",\n",
    "                                      m, \n",
    "                                      \"/\",\n",
    "                                      metadata_map[[m]]))\n",
    "        }        \n",
    "    }\n",
    "    \n",
    "    profile <- read_tsv(str_c(\"/g/scb2/zeller/karcher/dysb_classif/data/siamcat/profiles/\", file),\n",
    "                       skip = 61)\n",
    "    data[[length(data) + 1]] <- list(profile, metadata)\n",
    "    names(data)[length(data)] <- base\n",
    "\n",
    "    print(\"################################################\")\n",
    "    print(\"################################################\")\n",
    "    print(\"################################################\")\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "id": "9fbaf8bb",
   "metadata": {},
   "outputs": [],
   "source": [
    "data <- map2(data, names(data), function(d, na) {\n",
    "    if (na == \"Zhu_2018_Microbiome\") {\n",
    "        tmp <- d[[2]]\n",
    "        tmp <- tmp %>%\n",
    "        mutate(Sample_ID = str_replace(Sample_ID, \"D$\", \"\"))\n",
    "        d[[2]] <- tmp\n",
    "        \n",
    "        return(d)\n",
    "    } else if (na == \"Qin_2012_Nature_2\") {\n",
    "        tmp <- d[[1]]\n",
    "        colnames(tmp) <- str_replace(colnames(tmp), \"bgi-\", \"\")\n",
    "        d[[1]] <- tmp\n",
    "        \n",
    "        return(d)\n",
    "    } else if (na == \"Suez_2018_Cell\") {\n",
    "        tmp <- d[[2]]\n",
    "        tmp <- tmp %>%\n",
    "        mutate(Sample_ID = Alias) %>%\n",
    "        select(-Alias)\n",
    "        d[[2]] <- tmp\n",
    "        \n",
    "        return(d)\n",
    "    } else if (na == \"Zhang_205_NatMed\") {\n",
    "        tmp <- d[[2]]\n",
    "        tmp <- tmp %>%\n",
    "        filter(str_detect(External_ID, \"stool\"))\n",
    "        d[[2]] <- tmp\n",
    "        \n",
    "        return(d)\n",
    "    } else {\n",
    "        return(d)\n",
    "    }\n",
    "})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8c46391a",
   "metadata": {},
   "source": [
    "## TODO\n",
    "\n",
    "- Fix mapping rate between metadata and profiles of Nielsen_2014_NatBiotechnol (is only ~40% right now).\n",
    "    - I'm actually kicking out this dataset for now because it has also duplicate rownames. Have to talk to Jakob about this\n",
    "- Get correct mapping table from Jakob at some point to fix Bedarf (is 0% right now).\n",
    "    - Kicking this out right now too"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "id": "054a4183",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[90m# A tibble: 32 x 3\u001b[39m\n",
      "   dataset                      FractionProfileInMeta FractionMetaInProfile\n",
      "   \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m                        \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m                 \u001b[3m\u001b[90m<fct>\u001b[39m\u001b[23m                \n",
      "\u001b[90m 1\u001b[39m Backhed_2015_CellHostMicrobe 0.997506              1.000000             \n",
      "\u001b[90m 2\u001b[39m Bedarf_2017_GenomeMed        0.000000              0.000000             \n",
      "\u001b[90m 3\u001b[39m Feng_2015_NatComm            0.993631              1.000000             \n",
      "\u001b[90m 4\u001b[39m Franzosa_NatMed_2019         0.995475              1.000000             \n",
      "\u001b[90m 5\u001b[39m He_2017_GigaScience          0.943089              0.991453             \n",
      "\u001b[90m 6\u001b[39m Hoyles_2018_NatMed           0.986486              1.000000             \n",
      "\u001b[90m 7\u001b[39m Jie_2017_NatComm             0.997409              0.950617             \n",
      "\u001b[90m 8\u001b[39m Karlsson_2013_Nature         0.993151              1.000000             \n",
      "\u001b[90m 9\u001b[39m Kushugolova_2018_BMJOpen     0.994152              1.000000             \n",
      "\u001b[90m10\u001b[39m LeChatelier_2013_Nature      0.996587              1.000000             \n",
      "\u001b[90m11\u001b[39m Lewis_2015_CellHostMicrobe   0.997297              1.000000             \n",
      "\u001b[90m12\u001b[39m Li_2017_Microbiome           0.994924              1.000000             \n",
      "\u001b[90m13\u001b[39m Lloyd-Price_2019_Nature      0.999241              1.000000             \n",
      "\u001b[90m14\u001b[39m Loomba_2017_CellMetab        0.988506              1.000000             \n",
      "\u001b[90m15\u001b[39m Nielsen_2014_NatBiotechnol   0.445844              0.446970             \n",
      "\u001b[90m16\u001b[39m Palleja_2019_NatMicrobiol    0.982759              1.000000             \n",
      "\u001b[90m17\u001b[39m Poyet_2019_NatMedicine       0.998227              1.000000             \n",
      "\u001b[90m18\u001b[39m Qin_2012_Nature_2            0.800539              1.000000             \n",
      "\u001b[90m19\u001b[39m Qin_2014_Nature              0.995798              1.000000             \n",
      "\u001b[90m20\u001b[39m Schirmer_2016_Cell           0.997881              1.000000             \n",
      "\u001b[90m21\u001b[39m Suez_2018_Cell               0.999380              1.000000             \n",
      "\u001b[90m22\u001b[39m Thomas_2019_NatMedicine      0.992908              1.000000             \n",
      "\u001b[90m23\u001b[39m Vincnet_2016_Microbiome      0.995652              1.000000             \n",
      "\u001b[90m24\u001b[39m Wen_2017_GenomeBio           0.995283              1.000000             \n",
      "\u001b[90m25\u001b[39m Wirbel_2019_NatMedicine      0.991736              1.000000             \n",
      "\u001b[90m26\u001b[39m Xie_2016_CellSyst_2          0.996016              0.984252             \n",
      "\u001b[90m27\u001b[39m Yachida_2019_NatMedicine     0.934985              1.000000             \n",
      "\u001b[90m28\u001b[39m Yu_2017_Gut                  0.992248              1.000000             \n",
      "\u001b[90m29\u001b[39m Zeevi_2015_Cell              0.999154              0.999154             \n",
      "\u001b[90m30\u001b[39m Zeller_2014_MSB              0.993631              1.000000             \n",
      "\u001b[90m31\u001b[39m Zhang_205_NatMed             0.436911              1.000000             \n",
      "\u001b[90m32\u001b[39m Zhu_2018_Microbiome          0.955224              0.962406             \n"
     ]
    }
   ],
   "source": [
    "# Some diagnostics of how many samples we can map between metadata and profiles!\n",
    "tmp <- map(data, function(x) {\n",
    "   if (\"Sample_ID\" %in% colnames(x[[2]])) {\n",
    "       a <- mean( colnames(x[[1]]) %in% x[[2]]$Sample_ID )\n",
    "       b <- mean(x[[2]]$Sample_ID %in% colnames(x[[1]]) )    \n",
    "       return(c(a, b))\n",
    "   } else {\n",
    "       return(\"wrong colname...\")\n",
    "   }\n",
    "   a <- mean( colnames(x[[1]]) %in% x[[2]]$Sample_ID )\n",
    "   #b <- sum( colnames(x[[1]]) %in% rownames(x[[2]]) )\n",
    "    return(a)\n",
    "})\n",
    "\n",
    "# Kicking out Nielsen_*\n",
    "data <- data[which(names(data) != \"Nielsen_2014_NatBiotechnol\")]\n",
    "# Kicking out Bedarf\n",
    "data <- data[which(names(data) != \"Bedarf_2017_GenomeMed\")]\n",
    "\n",
    "data.frame(dataset = names(tmp),\n",
    "                FractionProfileInMeta = map_chr(tmp, function(x) x[1]), \n",
    "                FractionMetaInProfile = map_chr(tmp, function(x) x[2])) %>% as_tibble()  %>% print(n=100)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1a609757",
   "metadata": {},
   "source": [
    "## adapt metadata for one-pass classifier\n",
    "\n",
    "In order to use the metadata, I need a consistent target label. For this I'm going through the metadata I have manually and create a case/control column."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "193a61d0",
   "metadata": {},
   "source": [
    "## TODO: Discuss with Jakob.\n",
    "- For Hoyles_2018_NatMed, ask Jakob what are cases and what are controls\n",
    "    - in /g/scb2/zeller/SHARED/DATA/metadata/Hoyles_2018_NatMed/meta_hoyles.tsv it says Fibrosis stage 0 to stage X (I think stage 5?)\n",
    "- Same for Loomba_2017_CellMetab. Here we have Fibrosis of several stages and it's not clear which are 'cases' and which are 'controls'..\n",
    "- For Palleja_2019_NatMicrobiol I'm not sure what to do since it's not case controls but instead ABX-treated people sampled over time. Maybe take timepoint 0 as CTRs and ditch the rest?\n",
    "- Poyet_2019_NatMedicine: MetaG accompanying Eric Alm's isolate collection. Take earliest time point of each individual\n",
    "- Schirmer_2016_Cell: All samples marked as CTR, yet paper talks about assoc. between dysbiosis and cytokines. Exclude for now, talk to Jakob what he did.\n",
    "- Suez_2018_Cell: Overwhelming number of levels in Group, not clear what to do here. Talk to Jakob.\n",
    "- Backhed_2015_CellHostMicrobe: Mother/Infant pairs. We can try to take only mothers for controls; althrough right now we're not making use of cohort studies. So throw out for now\n",
    "- Vincnet_2016_Microbiome: Take earliest time point of each CTR individual.\n",
    "- Xie_2016_CellSyst_2: I'm keeping this cohort as-is (so only CTRs). Discuss with Jakob what he thinks.\n",
    "- Zeevi: I'm keeping as is"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "id": "71fc7c41",
   "metadata": {},
   "outputs": [],
   "source": [
    "data <- data[-which(names(data) %in% c(\"Hoyles_2018_NatMed\",\n",
    "                                       \"Loomba_2017_CellMetab\",\n",
    "                                       \"Palleja_2019_NatMicrobiol\",\n",
    "                                       \"Schirmer_2016_Cell\",\n",
    "                                        \"Suez_2018_Cell\",\n",
    "                                      \"Backhed_2015_CellHostMicrobe\"))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "id": "fe9ba81d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Feng_2015_NatComm\"\n",
      "[1] \"Franzosa_NatMed_2019\"\n",
      "[1] \"He_2017_GigaScience\"\n",
      "[1] \"Jie_2017_NatComm\"\n",
      "[1] \"Karlsson_2013_Nature\"\n",
      "[1] \"Kushugolova_2018_BMJOpen\"\n",
      "[1] \"LeChatelier_2013_Nature\"\n",
      "[1] \"Lewis_2015_CellHostMicrobe\"\n",
      "[1] \"Li_2017_Microbiome\"\n",
      "[1] \"Lloyd-Price_2019_Nature\"\n",
      "[1] \"Poyet_2019_NatMedicine\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“`cols` is now required when using unnest().\n",
      "Please use `cols = c(data)`”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Qin_2012_Nature_2\"\n",
      "[1] \"Qin_2014_Nature\"\n",
      "[1] \"Thomas_2019_NatMedicine\"\n",
      "[1] \"Vincnet_2016_Microbiome\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“`cols` is now required when using unnest().\n",
      "Please use `cols = c(data)`”\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Wen_2017_GenomeBio\"\n",
      "[1] \"Wirbel_2019_NatMedicine\"\n",
      "[1] \"Xie_2016_CellSyst_2\"\n",
      "[1] \"Yachida_2019_NatMedicine\"\n",
      "[1] \"Yu_2017_Gut\"\n",
      "[1] \"Zeevi_2015_Cell\"\n",
      "[1] \"Zeller_2014_MSB\"\n",
      "[1] \"Zhang_205_NatMed\"\n",
      "[1] \"Zhu_2018_Microbiome\"\n"
     ]
    }
   ],
   "source": [
    "dataMetaAdapted <- list()\n",
    "for (dataset in names(data)){\n",
    "    print(dataset)\n",
    "    tmp <- data[which(names(data) == dataset)][[1]]\n",
    "    if (dataset == \"Backhed_2015_CellHostMicrobe\") {\n",
    "        # Swedish mother/infant pairs. \n",
    "        # I'm keeping both infants and mothers as controls\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = \"control\") %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Feng_2015_NatComm\"){\n",
    "        # Controls/adenoma/CRC. \n",
    "        # I'm keeping only controls and CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        #filter(Group == \"CTR\" | Group == \"CRC\") %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"CTR\") {\n",
    "                return(\"ADA/CRC\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"Franzosa_NatMed_2019\"){\n",
    "        # Controls/UC/CD\n",
    "        # I'm keeping everything\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        #filter(Group == \"CTR\" | Group == \"CRC\") %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"CTR\") {\n",
    "                return(\"UC/CD\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"He_2017_GigaScience\"){\n",
    "        # CD, CD-treated, controls\n",
    "        # I'm keeping CD and controls.\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group != \"CD-treated\") %>%\n",
    "        filter(Group == \"CTR\" | Group == \"CD\") %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"IBD\")\n",
    "    } else if (dataset == \"Hoyles_2018_NatMed\"){\n",
    "        # TODO!\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(condition = \"No condition\")  \n",
    "    } else if (dataset == \"Jie_2017_NatComm\"){\n",
    "        # Controls, ACVD\n",
    "        # I'm keeping everything\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"ACVD\")\n",
    "    } else if (dataset == \"Karlsson_2013_Nature\"){\n",
    "        # Controls, IGT, T2D\n",
    "        # I'm keeping Controls and T2D\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group == \"CTR\" | Group == \"T2D\") %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"T2D\")\n",
    "    } else if (dataset == \"Kushugolova_2018_BMJOpen\"){\n",
    "        # Controls, MS\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        #filter(Group == \"CTR\" | Group == \"T2D\") %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"MS\")\n",
    "    } else if (dataset == \"LeChatelier_2013_Nature\"){\n",
    "        # Controls, OB\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"OB\")\n",
    "    } else if (dataset == \"Lewis_2015_CellHostMicrobe\"){\n",
    "        # Controls, CD\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"IBD\")\n",
    "    } else if (dataset == \"Li_2017_Microbiome\"){\n",
    "        # Controls, pre-hypertension, HTN\n",
    "        # I'm keeping controls and HTN\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group %in% c(\"CTR\", \"HTN\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"HTN\")\n",
    "    } else if (dataset == \"Li_2017_Microbiome\"){\n",
    "        # Controls, pre-hypertension, HTN\n",
    "        # I'm keeping controls and HTN\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group %in% c(\"CTR\", \"HTN\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"HTN\")\n",
    "    } else if (dataset == \"Lloyd-Price_2019_Nature\"){\n",
    "        # Controls,CD, UC\n",
    "        tmp[[2]] <- tmp[[2]] %>%        \n",
    "        #filter(Group %in% c(\"CTR\", \"HTN\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"nonIBD\", 'control', 'case')) %>%\n",
    "        #mutate(condition = \"IBD\")\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"nonIBD\") {\n",
    "                return(\"UC/CD\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"Loomba_2017_CellMetab\"){\n",
    "        # TODO!\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(condition = \"No condition\")    \n",
    "    } else if (dataset == \"Palleja_2019_NatMicrobiol\"){\n",
    "        # TODO!\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(condition = \"No condition\")  \n",
    "    } else if (dataset == \"Poyet_2019_NatMedicine\"){\n",
    "        # Longitudinally sampled people.\n",
    "        # I'm taking the first time point per individual.\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        group_by(Individual_ID) %>%\n",
    "        nest() %>%\n",
    "        mutate(data = map(data, function(x) {\n",
    "                          return(x %>%\n",
    "                          mutate(Timepoint = as.numeric(as.character(Timepoint))) %>%\n",
    "                          arrange(Timepoint) %>%\n",
    "                         head(1))\n",
    "            })) %>%\n",
    "        unnest() %>%\n",
    "        mutate(caseControls = \"control\") %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Qin_2012_Nature_2\"){\n",
    "        # Controls, T2D\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        #filter(Group %in% c(\"CTR\", \"HTN\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"T2D\")\n",
    "    } else if (dataset == \"Qin_2014_Nature\"){\n",
    "        # Controls, Liver cirrhosis\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"LCIR\")\n",
    "    } else if (dataset == \"Schirmer_2016_Cell\"){\n",
    "        # TODO\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Suez_2018_Cell\"){\n",
    "        # TODO\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Thomas_2019_NatMedicine\"){\n",
    "        # Controls, adenoma, CRC\n",
    "        # I'm keeping controls and CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group %in% c(\"CTR\", \"CRC\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"CTR\") {\n",
    "                return(\"ADA/CRC\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"Vincnet_2016_Microbiome\"){\n",
    "        # Typo above is on purpose\n",
    "        # Controls, CDI\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group == \"CTR\") %>%\n",
    "        group_by(Individual_ID) %>%\n",
    "        nest() %>%\n",
    "        mutate(data = map(data, function(x) {\n",
    "            return(x %>%\n",
    "                          mutate(Timepoint = as.numeric(as.character(Timepoint))) %>%\n",
    "                          arrange(Timepoint) %>%\n",
    "                         head(1))\n",
    "            })) %>%\n",
    "        unnest() %>%\n",
    "        mutate(caseControls = \"control\") %>%\n",
    "        mutate(condition = \"CDI\")\n",
    "    } else if (dataset == \"Wen_2017_GenomeBio\"){\n",
    "        # Controls, AS\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = 'AS')\n",
    "    } else if (dataset == \"Wirbel_2019_NatMedicine\"){\n",
    "        # Controls, CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = \"CRC\")\n",
    "    } else if (dataset == \"Xie_2016_CellSyst_2\"){\n",
    "        # Controls (this is TwinsUK)\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = \"control\") %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Yachida_2019_NatMedicine\"){\n",
    "        # Controls, Adenoma, CRC, ...\n",
    "        # I'm keeping Controls and CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group %in% c(\"CTR\", \"CRC\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"CTR\") {\n",
    "                return(\"ADA/CRC\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"Yu_2017_Gut\"){\n",
    "        # Controls, CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        #filter(Group %in% c(\"CTR\", \"CRC\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = 'CRC')\n",
    "    } else if (dataset == \"Zeevi_2015_Cell\"){  \n",
    "        # Controls\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = \"control\") %>%\n",
    "        mutate(condition = \"No condition\")\n",
    "    } else if (dataset == \"Zeller_2014_MSB\"){  \n",
    "        # Controls, Adenoma, CRC, ...\n",
    "        # I'm keeping Controls and CRC\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        filter(Group %in% c(\"CTR\", \"CRC\")) %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = map_chr(Group, function(g) {\n",
    "            if (g == \"CTR\") {\n",
    "                return(\"ADA/CRC\")\n",
    "            } else {\n",
    "                return(g)\n",
    "            }\n",
    "        }))\n",
    "    } else if (dataset == \"Zhang_205_NatMed\"){ \n",
    "        # Controls, Rheumatoid arthritis.\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = 'RA')\n",
    "    } else if (dataset == \"Zhu_2018_Microbiome\"){\n",
    "        # Controls, breast cancer\n",
    "        tmp[[2]] <- tmp[[2]] %>%\n",
    "        mutate(caseControls = ifelse(Group == \"CTR\", 'control', 'case')) %>%\n",
    "        mutate(condition = 'BC')         \n",
    "    } else {\n",
    "        print(\"We re not having a fix for dataset: \")\n",
    "        print(dataset)\n",
    "        asdsadsad\n",
    "    }\n",
    "    dataMetaAdapted[[length(dataMetaAdapted) + 1]] <- tmp\n",
    "    names(dataMetaAdapted)[length(dataMetaAdapted)] <- dataset\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "id": "a2fe52a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataMetaAdaptedTrunc <- map2(dataMetaAdapted, names(dataMetaAdapted), function(d, na) {\n",
    "    d[[1]] <- d[[1]] %>%\n",
    "    as.data.frame()\n",
    "    rownames(d[[1]]) <- d[[1]]$X1\n",
    "    d[[1]]$X1 <- NULL    \n",
    "    sharedSamples <- intersect(colnames(d[[1]]), d[[2]]$Sample_ID)\n",
    "    d[[1]] <- d[[1]][colnames(d[[1]]) %in% sharedSamples]\n",
    "    d[[2]] <- d[[2]] %>%\n",
    "    filter(Sample_ID %in% sharedSamples)\n",
    "    d[[2]] <- d[[2]] %>%\n",
    "    as.data.frame() %>%\n",
    "    distinct()\n",
    "    rownames(d[[2]]) <- d[[2]]$Sample_ID\n",
    "    d[[2]]$Sample_ID <- NULL\n",
    "    return(d)\n",
    "})\n",
    "dimens <- map(dataMetaAdaptedTrunc, function(x) return(c(dim(x[[1]])[2], dim(x[[2]])[1])))\n",
    "stopifnot(all(map_dbl(dimens, function(x) x[[1]]) == map_dbl(dimens, function(x) x[[2]])))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "id": "15acf36d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cast profile to long, merge with metadata table\n",
    "dataMetaAdaptedTruncLong <- map2(dataMetaAdaptedTrunc, names(dataMetaAdaptedTrunc), function(x, ds) {\n",
    "    return(x[[1]] %>%\n",
    "    as.data.frame() %>%\n",
    "    mutate(taxon = rownames(.)) %>%\n",
    "    pivot_longer(cols = -taxon) %>%\n",
    "    rename(Sample_ID = name,\n",
    "          count = value) %>%\n",
    "    left_join(x[[2]] %>%\n",
    "              mutate(Sample_ID = rownames(.)), by = \"Sample_ID\") %>%\n",
    "    mutate(dataset = ds))\n",
    "})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "id": "1b565542",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Select columns you're interested in\n",
    "dataMetaAdaptedTruncLongAll <- rbindlist(map(dataMetaAdaptedTruncLong, function(x) {\n",
    "    return(x %>%\n",
    "    select(taxon, Sample_ID, count, caseControls, dataset, condition))\n",
    "}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "id": "7286427f-4366-4c83-80eb-5c850173ff1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` ungrouping output (override with `.groups` argument)\n",
      "`summarise()` ungrouping output (override with `.groups` argument)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 425   2\n"
     ]
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA0gAAANICAMAAADKOT/pAAACqVBMVEUAAAABAQECAgIDAwME\nBAQFBQUGBgYHBwcJCQkKCgoLCwsODg4PDw8QEBARERESEhITExMUFBQXFxcYGBgZGRkaGhob\nGxscHBwdHR0eHh4fHx8gICAhISEiIiIjIyMkJCQlJSUmJiYnJycoKCgrKyssLCwtLS0uLi4v\nLy8wMDAxMTEzMzM0NDQ5OTk7Ozs9PT0+Pj5AQEBBQUFCQkJDQ0NERERFRUVGRkZHR0dISEhJ\nSUlKSkpLS0tMTExNTU1OTk5PT09QUFBRUVFSUlJTU1NUVFRVVVVWVlZXV1dZWVlaWlpbW1tc\nXFxdXV1eXl5fX19gYGBhYWFiYmJjY2NkZGRlZWVmZmZnZ2doaGhpaWlqampra2tsbGxtbW1v\nb29wcHBxcXFycnJ0dHR1dXV2dnZ3d3d4eHh5eXl6enp7e3t8fHx9fX1+fn5/f3+AgICBgYGC\ngoKDg4OEhISFhYWGhoaHh4eIiIiKioqLi4uMjIyPj4+QkJCRkZGTk5OUlJSVlZWWlpaXl5eY\nmJiZmZmbm5ucnJyenp6fn5+goKCjo6OlpaWmpqanp6eoqKipqamrq6usrKytra2urq6vr6+w\nsLCxsbGysrK0tLS1tbW2tra3t7e4uLi5ubm6urq7u7u8vLy9vb2+vr6/v7/AwMDBwcHCwsLD\nw8PExMTFxcXGxsbHx8fIyMjJycnKysrLy8vMzMzNzc3Ozs7Pz8/Q0NDR0dHS0tLT09PU1NTV\n1dXW1tbX19fY2NjZ2dna2trb29vc3Nzd3d3e3t7f39/g4ODh4eHi4uLj4+Pk5OTl5eXm5ubn\n5+fo6Ojp6enq6urr6+vs7Ozt7e3u7u7v7+/w8PDx8fHy8vLz8/P09PT19fX29vb39/f4+Pj5\n+fn6+vr7+/v8/Pz9/f3+/v7///9JGxcdAAAACXBIWXMAABJ0AAASdAHeZh94AAAgAElEQVR4\nnO3d/4NddX3n8bvdFbXb1q0Iu+jqrlvFpdN2t92yrKvMFEiRgWgSQktpqQtUkNBVTLGskLam\nQLbGul0tuLaSTVMpKYEYx4YVCRKiYrGWEJKZyXyfe+/nL9l778zN3PO65977Pu/PmXsm9z6f\nP8yZSWY+73M+5zwydwJiKRBRdKWiT4BoEAISUQ4BiSiHgESUQ0AiyiEgEeUQkIhyCEhEORQP\n6bSlhcqk6fO6thy/xGRlPn6R2Zn4NeY3zIYsxC8yMxu/xnxlKn6RHDbkTKYNmcwR0muWFsLr\nps/rWiV+idNhPn6RmZn4NRbC6fhFctmQhfhFzuawIfMbZENeD4sZPvs0kCICkgQkIHkCkgQk\nIHkCkgQkIHkCkgQkIHkCkgQkIHkCkgQkIHkCkgQkIHkCkgSkPCCdtbQcZkyf17Vq/BKzYSl+\nkYWF+DWWw2z8IrlsyHL8IguL8WssbZANmcm0ITM5QpqzVA7zps/rWjV+iflQjl9kaSl+DTZE\nWg4L8Yv0f0NyhGT6FshLO4mXdhIv7UwDgSQBSQKSaSCQJCBJQDINBJIEJAlIpoFAkoAkAck0\nEEgSkCQgmQYCSQKSBCTTQCBJQJKAZBoIJAlIEpBMA4EkAUkCkmkgkCQgSUAyDQSSBCQJSKaB\nQJKAJAHJNBBIEpAkIJkGAkkCkgQk00AgSUCSgGQaCCQJSBKQTAOBJAFJApJpIJAkIElAMg0E\nkgQkCUimgUCSgCQByTQQSBKQJCCZBgJJApIEJNPAnpDGV+r6OUCSgCQBCUiegCQBCUiegCQB\nCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQB\nCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQB\nCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQBCUiegCQN\nM6TxRF0XAZIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlI\nQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlI\nQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJAlIQPIEJGnDQ3pubDqEyt7tW/eU145AighI\n0lBAmt0+WoO0d8vhie0PrR2BFBGQpKGA9Jnba5DmrjsYwpFNk80jkGICkjQMkL5287dqkF4c\nnQlhaexo81j7jc9u3rz5lmVL1dDpd5KQui7ScQ175VCJX6SSwxrVUI5fJIcNWQ7V+DXy2JDK\nebkhS1kg/eP1Lx6vQTp0df2D8QPNY+3NjpGRkcst39K6lIQUuRhRX6uce683pMrH/izUIT11\nTf2j8X3NY/P3Td8CeWkn8dJOGvyXdo/+xsuvPDX6wuljo7MhlMcmmkcgxQQkafAh7R5ttGv2\n2mdCePaqM80jkGICkjT4kOrVX9qFh28+fuLWXWtHIEUEJGmIIJUf2bZl9/LaEUgRAUkaDkjd\nMg0EkgQkCUimgUCSgCQByTQQSBKQJCCZBgJJApIEJNNAIElAkoBkGggkCUgSkEwDgSQBSQKS\naSCQJCBJQDINBJIEJAlIpoFAkoAkAck0EEgSkCQgmQYCSQKSBCTTQCBJQJKAZBoIJAlIEpBM\nA4EkAUkCkmkgkCQgSUAyDQSSBCQJSKaBQJKAJAHJNBBIEpAkIJkGAkkCkgQk00AgSUCSgGQa\nCCQJSBKQTAOBJAFJApJpIJAkIElAMg0EkgQkCUimgUCSgCQByTQQSBKQJCCZBgJJApIEJNNA\nIElAkoBkGggkCUgSkEwDgSQBSQKSaSCQJCBJQDINBJIEJAlIpoFAkoAkAck0EEgSkCQgmQYC\nSQKSBCTTQCBJQJKAZBoIJAlIEpBMA4EkAUkCkmkgkCQgSUAyDQSSBCQJSKaBQJKAJAHJNBBI\nEpAkIJkGAkkCkgQk00AgSUCSgGQaCCQJSBKQTAOBJAFJApJpIJAkIElAMg0EkgQkCUimgUCS\ngCQByTQQSBKQJCCZBgJJApIEJNPAFEjjaXVdBEgSkCQgAckTkCQgAckTkCQgAckTkCQgAckT\nkCQgAckTkCQgAckTkCQgmTgBSQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQg\neQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSdF5D\nmrS0FKb1l7pA6rBIxTSqa9NhMX6R+fn4NVI2JHu5bMhS/CJzOWzIYjgbv0gOGzKVaUOmc4Q0\nb6kcFvSXukDqsEjVNKprC6Ecv8jycvwaKRuSvY2yIUsDtSGVLJ+eIyTTt0Be2km8tJN4aWca\nCCQJSBKQTAOBJAFJApJpIJAkIElAMg0EkgQkCUimgUCSgCQByTQQSBKQJCCZBgJJApIEJNNA\nIElAkoBkGggkCUgSkEwDgSQBSQKSaSCQJCBJQDINBJIEJAlIpoFAkoAkAck0EEgSkCQgmQZm\ng9TBE5AkIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApIn\nIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIElAApInIEkbG9IPPjm++f6T\nIVT2bt+6p7x2BFJEQJIGH9LiTb///OGP3RbC3i2HJ7Y/tHYEUkRAkgYf0gujUyEcHZ2fu+5g\nCEc2TTaPQIoJSNLgQ6pWwtKrD94RXhydCWFp7GjzWPutL9x5552fWrBUCYv6S70h6VdUTaO6\nthjK8YssL8evUW7fkOzlsiGV+EWGeEMyQKp15+gNr4dDV9ffHT/QPNbe7BgZGbncskBavSF5\nVybqU5Vz75kgTb38R1tnn7qm/u74vuax9ub1V1555R9OW1oMk/pLvSHpV1RMo7o2FRbjF5md\njV8jZUOyl8OGTG6UDVkIU/GL5LIhS5k+2w7p5Pdqb8qbnj42Ols7jk00j83fN72W5GckiZ+R\npMH/GenJj1RDmP+VI7PXPhPCs1edaR6BFBOQpMGHNPWhP3zp2L03zYeHbz5+4tZd4dwRSBEB\nSRp8SOH5T9zw4ft/VHs598i2LbuX145AighI0hBA6pFpIJAkIElAMg0EkgQkCUimgUCSgCQB\nyTQQSBKQJCCZBrogqScgSUCSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CS\ngAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CS\ngAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CS\ngAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CSgAQkT0CShgRSEgaQogOSBCQg\neQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQg\neQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQg\neQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSBCQgeQKSNOiQUmEAKTogSUACkicg\nSUACkicgSUACkicgSUACkicgSUACkicgSUACkicgSUACkicgSUACkicgSUACkicgSUDKDEl/\nz3zhQEoEJAlIxgsHUiIgScVBOtO7JIzUX+zSuWUq7UsahieaDotZv6S9ubn4NRbDVPwild6f\n0qupXDZkPn6NxTAdv0gOGzIZljJ89lSOkJZ6l4SR+otdSn5+6kfmlkMl65e0Vy7Hr1EJy/GL\nVOOXYEO0TBuymCMkwzfAJIzUX+xS8vNTP8rwrZiXdol4aScN/s9IQGoLSBKQgOQJSBKQgOQJ\nSBKQgOQJSBKQgOQJSBKQgOQJSBKQgOQJSBKQgOQJSBKQgOQJSBKQgOQJSBKQ7CW/OtsmAakt\nIElAMl44kBIBSQKS8cKBlAhIEpCMFw6kRECSgGS8cCAlApIEJOOFAykRkCQgGS8cSImAJAHJ\neOFASgQkCUjGCwdSIiBJQDJeOJASAUkCkvHCgZQISBKQjBcOpERAkoBkvHAgJQKSBCTjhQMp\nEZAkIBkvHEiJgCQByXjhQEoEJAlIxgsHUiIgSUAyXjiQEgFJApLxwoGUCEhSPpA2H1s5Pvlb\nQOockCQgJSCdPXWq9OVT9U7e/WYgdQ5IEpASkLaW1vpPQOockCQgJSB99YEHSrc80OgPXgZS\n54AkAUl/RvrFo3ZAQIoJSNJgQfJkGNcvSL11AUkCkpQPpKntl/z0SkDqHJAkICmkm0o/d+NN\njYDUOSBJQFJIb/2QHRCQYgKSNFiQ3rwHSL0DkgQkhfSfbwVS74AkAUkh/f17di8DqVdAkoCk\nkO79hdKbfuayekDqHJAkICmkHe9vBqTOAUkCEv9A1hOQJCCdh5BS10ou2fXCgZQISFI+kN7b\n7FNA6hyQJCAppCvrXfHO0n/8IpA6ByQJSB1e2n31rV8HUueAJAGp089Id2/cv7UD0lpAkjYc\npP/5U0DqHJAkIHWAVB79V0DqHJAkIKX+ZcOVH3xX6XYgdQ5IEpAU0mUr/fw9C0DqHJAkIBXy\nD2TzLHVO1wsHUiIgSXlBqn7/wOPfqwKpW0CSgNQG6cDP1v+rdu87cJ5B6qZLLhxIiYAk5QPp\nmxdcdO+fP7bzojdm+M9yGcYBqS0gSYMF6QNvf61h4+1XAqlzQJKApJAu/PjKccfbgNQ5IElA\navuvCDUhXQikzgFJApJC+q+XNF7anXrHB4DUOSBJQFJIExdctPOxx+67+IIJIHUOSBKQ2v76\n+6/eV//r70v32x0ByRWQpAGDFKon9u17qZLBEZBcAUkaMEindz0Rwh9/ZhJIXQKSBCSFdOod\npd0hfLT09h8AqXNAkoCkkLb9xF/VD3/7U5uB1DkgSUBSSO+6Y+V413r+D/uA9BqQ2hosSG/5\n5Mpx53r+T82B9BqQ2hosSFe8d7Z+mP/3vwykzgFJApJCevoN79vz9JG9Iz/210DqHJAkILX9\n9fdfvLv+D2Qv/oLdEZBcAUkaMEhh+euf/5ODcxkcAckVkKRBg5Q9wzggtQUkCUiGcesHqZuu\n9AsHUiIgSesIaf6zv3bdPd8NobJ3+9Y95bUjkCICkjQEkO6+5Rvfuf+G02HvlsMT2x8K545A\nighI0uBDemX0uRDK1z8+d93BEI5smmwegRQTkKTBh/Tqny6GsPCrf/ni6EwIS2NHm8fabz12\n3333PTjfuz5D6nImC6FsON8eLS/Hr1EOC/GLVOOXyGVDlgZqQypZPj0DpHoL9287e+jq+nvj\nB5rH2psdIyMjlxu+vM+QTJdEFN/a/3LPAql6YNtdk+Gpa+rvj+9rHmtvfvj8889/50zv+gyp\ny5lMh0XD+fZobi5+jcUwFb9IJX6JqTw2ZDaXDZmOXySHDZkMSxk+eyoLpMmP3/hENYRjo7O1\nn5XGJprH5m8bXkn2GVLX17T8jJSIn5Gk9fsZqXrbzsa/yTp77TMhPHvVmeYRSDEBSRp8SEfH\nnjha67Xw8M3HT9y6K5w7AikiIEmDD+nR0UZfCeVHtm3ZvRzOHYEUEZCkwYfUK8M4ILUFJAlI\nhnFAagtIEpAM44DUFpAkIBnGAaktIElAMowDUltAkoBkGAektoAkAckwDkhtAUkCkmFcMZDS\nWAFJApIEJCB5ApIEJCB5ApIEJCB5ApIEpB41LxxIiYAkAalHzQsHUiIgSUDqUfPCgZQISBKQ\netS8cCAlApIEpB41LxxIiYAkAalHzQsHUiIgSUDqUfPCgZQISBKQetS8cCAlApIEJFtASgYk\nCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckWkJIBSQKSLSAlA5IEJFtA\nSgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckWkJIBSQKSLSAlA5IE\nJFtASgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckWkJIBSQKSLSAl\nA5IEJFtASgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckWkJIBSQKS\nLSAlA5IEJFtASgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckWkJIB\nSQKSLSAlA5IEJFtASgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAkAckW\nkJIBSQKSLSAlA5IEJFtASgYkCUi2gJQMSBKQbAEpGZAkINkCUjIgSUCyBaRkQJKAZAtIyYAk\nAckWkJIBSSoO0qneFa2nPcNJd252JurLGy2EM/GLVOKXqEGKX2RmNn6NjbMhixk++0yOkJZ7\nVzSb9gwn3blKJerLG1VDOX4Ry973XKMav0YeG1I5LzdkKUdIhm+ARbNpz/uNvxEv7SRe2gHJ\nE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJ\nE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJE5AkIAHJ\nE5AkIAHJE5AkIAHJE5AkIA0tpNV8ew0kCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQk\nCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQk\nCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQkCUhA8gQk\nCUjDDmm1jHsNJAlIQGqUca+BJAEJSI0y7jWQJCABqVHGvQaSBCQgNcq410CSgASkRhn3GkgS\nkIDUKONeA0kCEpAaZdxrIElAAlKjjHsNJAlIQGqUca+BJAEJSI0y7jWQJCABqVHGvQaSBCQg\nNcq410CSgASkRhn3GkgSkIDUKONeA0kCEpAaZdxrIElAAlKjjHsNJAlIQGqUca+BJAEJSI0y\n7jWQJCABqVHGvQaSBCQgNcq410CSgASkRhn3GkgSkIDUKONeA0kCEpAaZdxrIElAAlKjjHsN\nJAlIQGrNuntAkoAEpNasuwckCUhAas26e0CSgASk1qy7ByQJSFZITy4CaS0gSUAyQvrR2HTt\nbWXv9q17ymtHIEUEJGkIIJ34/LbROqS9Ww5PbH9o7QikiIAkDQGkw7s/XYc0d93BEI5smmwe\ngRQTkKQhgBTC8TqkF0dnQlgaO9o81n79bz73uc/9r5neFS3EmOFKGi0uWj+zc8thLn6RavwS\ns2E5fpHB2pByps/ODunQ1fV3xw80j7U3O0ZGRi43fH3RQoxZtoKopcq59+yQnrqm/u74vuax\n9ualw4cPT0z2rmghxgxX0mh+3vqZnVsK0/GLVOOXmA5L8YvM5bAhi+Fs/CI5bMhUpg2ZdkA6\nNlr7PlYem2gem79peCVZtBBj1hfG/Iwk8TNSFkiz1z4TwrNXnWkegRQTkKThgRQevvn4iVt3\nrR2BFBGQpCGCVH5k25bdy2tHIEUEJGkoIHXNMK5oIcasuwckCUhAas26e0CSgASk1qy7ByQJ\nSEBqzbp7QJKABKTWrLsHJAlIQGrNuntAkoAEpNasuwckCUhAas26e0CSgASk1qy7ByQJSEBq\nzbp7QJKABKTWrLsHJAlIQGrNuntAkoAEpNasuwckCUhAas26e0CSgASk1qy7ByQJSEDqXOfL\nAZIEJCB1rvPlAEkCEpA61/lygCQBCUid63w5QJKABKTOdb4cIElAAlLPUi4HSBKQgNSzlMsB\nkgQkIPUs5XKAJAEJSD1LuRwgSUACUs9SLgdIEpCA1LOUywGSBCQg9SzlcoAkAQlIPUu5HCBJ\nQAJSz1IuB0gSkIDUs5TLAZIEJCD1LOVygCQBCUg9S7kcIElAAlLPUi4HSBKQgNSzlMsBkgQk\nIPUs5XKAJAEJSD1LuRwgSUACUs9SLgdIEpCA1LOUywGSBCQg9SzlcoAkAQlIPUu5HCBJQAJS\nz1IuB0gSkIBkrfVygCQBCUjWWi8HSBKQgGSt9XKAJAEJSNZaLwdIEpCAZK31coAkAQlI1lov\nB0gSkIBkrfVygCQBCUjWWi8HSBKQgGSt9XKAJAEJSNZaLwdIEpCAZK31coAkAQlI1lovB0gS\nkIBkrfVygCQBCUjWWi8HSBKQgGSt9XKAJAEJSNZaLwdIEpCAZK31coAkAQlI1lovB0gSkICU\nscblAEkCEpAy1rgcIElAAlLGGpcDJAlIQPLlvUtrAUkCkmFc0c997nnv0lpAkoBkGFf0c597\n3ru0FpAkIBnGFf3c5573Lq0FJAlIhnFFP/e5571LawFJApJhXNHPfe5579JaQJKAZBhX9HOf\ne967tBaQJCAZxhX93Oee9y6tBSQJSIZxRT/3uee9S2sBSQKSYVzRz33uJa/LcbuApAHJMK7Q\nh349Sr2uTPcLSBKQDOOKedrXsdTrynS/gCQByTCumKd9HUu9rkz3C0gSkMq9K+ZpX8dSr8uw\nEWtVLfvWq1zWqMavUanEr1ENOSzS9w1ZzhGSwW0xT/s6lnpdmf7g4zuSxHckw7hinvZ1LPW6\nMt0vIElAMowr5mlfx1KvK9P9ApIEJMO4Yp72fpfpfgFJApJhXNGPeH/KdL+AJAHJMK7oR7w/\nZbpfQJKAZBhX9CPenzLdLyBJQDKMK/oR70+Z7heQJCAZxhX9iPenTPcLSBKQDOOKfsT7U6b7\nBSQJSIZxRT/i/SnT/QKSBCTDuKIf8f6U6X4BSQKSYVzRj3h/ynS/gCQByTCu6Ee8P2W6X0CS\ngGQYV/Qj3p8y3S8gSUAyjCv6Ee9Pme4XkCQgGcYV/Yj3M+MdAJIEJMO4oh/ufma8A0CSgGQY\nV/TD3c+MdwBIEpAM44p+uPuZ8Q4ASQKSYVzRD3c/M94BIElAMowr+uHuZ902oOVXgCQByTCu\n0Ce7z3XbgNSPIgKSBKTBqdsGpH4UEZAkIA1O3TYg9aOIgCQBaeBK3YDUjyICkgSkgSt1A1I/\nighIEpAGteQGpH4UEZAkIA1qyQ1I/SgiIElAGtSSG2D4KFNAkoA0qCU3wPBRpoAkAWlQS26A\n4aNMAUkC0qCW3IBu25Htdq8EJAlIg1pyAwzbkemmA0kC0qCW3ADDdmS66UCSgDSoJTfAsB2Z\nbjqQJCANaskNMGxHppsOJAlIw5FhOzLddCBJQBqODNuR6aYDSQLScGTYjkw3HUgSkIYjw3Zk\nuulAkoA0HBm2I9NNB5IEpOHIsB2ZbjqQJCDRaskd67GvQJKARKsld6zHvgJJAhKtltyxHvsK\nJAlItFrqjnXaVyBJQKLVUnes074CSQISrZa6Y532FUgSkGi11B3rtK9AkoBEq6XuWKd9BZIE\nJOpWp30FkgQk6lanfQWSBCQy1LavQJKARIba9hVIEpDIUNu+Vl6z/jsQXZ4EICUC0tC0tq9A\nkoBE9tb2FUgSkMhe6qY6HpnVJwFIiYA0NKVuquORWX0SgJQISENT6qY6HpnVJwFIiYA03Hmf\nGyBJQBruvM8NkCQg0bjDE5AkINE4kOKXABKNAyl+CSDROJDilwASjQMpfgkg0TiQ4pcAEo0r\nJMMvAkkCEo0DKX4JINFayX1PvRnnngQgJQISZerckwCkRECiTJ17EoCUCEiUqeStcTxxLQEJ\nSENb8tY4nriWgOSCVNm7feueMpDO75K3xvHEtQQkF6S9Ww5PbH8ISENStzu6+hGQPJDmrjsY\nwpFNk0Aajrrd0dWPgOSB9OLoTAhLY0dr7379S1/60lfO9q7QB4FyKfWOrn60sJDxbqcuNmt4\nknpUjV9iJizr5XX9bD+kQ1fX344fqL3ZMTIycnnmBYgGpcq597JDeuqa+tvxfcH+Hens2eUw\nE/knxdlc/ryZDUvxiyT/APa1HDbGH8Cz9T+AY8tjQ5Y2yIbMZNqQmO9Ix0ZnQyiPTTQ/Nr2W\nXAiv+16ytpbLf+p6Pn6RmRx+JFjYID8S8A9kpf79jDR77TMhPHvVGSDFBCRp+CCFh28+fuLW\nXec+NA0EkgQkaQghlR/ZtmX3MpCiApI0hJAk00AgSUCSgGQaCCQJSBKQTAOBJAFJApJpIJAk\nIElAMg0EkgQkCUimgUCSgCQByTQQSBKQJCCZBgJJApIEJNNAIElAkoBkGggkCUgSkEwDgSQB\nSQKSaSCQJCBJQDINBJIEJAlIpoFAkoAkAck0EEgSkCQgmQYCSQKSBCTTQCBJQJKAZBoIJAlI\nEpBMA4EkAUkCkmkgkCQgSUAyDQSSBCQJSKaBQJKAJAHJNBBIEpAkIJkGAkkCkgQk00AgSUCS\ngGQaCCQJSBKQTAOBJAFJApJpIJAkIElAMg0EkgQkCUiTlr516Pf2SfsAAAWSSURBVDXT53Vt\nKX6Jk4eej19kdjZ+jW9vmA15Ln6RPDbk2KFX4xfJYUNOHfpWhs+ezhGSqd8ZebU/g3r03ZF7\niz6FlX535AdFn0KjH458vOhTWGnnyEtFn0Kj10dud30dkIoJSBKQTAFJApIEJFNAkoAkAcnU\ns/sX+jOoRzP7v130Kaz03P7Zok+h0dz+/1f0Kaz07f1niz6FRov7j7q+rk+QiAY7IBHlEJCI\ncmjgIT25WPQZ0DDUF0iVvdu37in3Y1JbPxqbbjmBTsf1bv6zv3bdPd8t/jzCDz45vvn+kxvg\nREJ4rn5nCj+P/z1a66ocTqQvkPZuOTyx/aF+TJJOfH7b6HTLCXQ6rnd33/KN79x/w+nCz2Px\npt9//vDHbit+Q0KY3V6/M4Wfx657JyYmvpnDifQD0tx1B0M4smmyD6Okw7s/Xb9dzRPodFzv\n03hl9LkQytc/XvR5hBdGp0I4Ojpf+ImE8Jnba3em+PP4nf/TOMSfSD8gvTg6E8LSmO/v5yM7\nXofUPIFOx/U+iVf/tPaD2sKv/mXR5xGqlbD06oN3FL4hIXzt5m/V7kzx5zH+qa3jO/8hhxPp\nB6RDV9ffjh/ow6i2GpCaJ9Dp2I8TWbh/29mNcB53jt7wevEb8o/Xv1i/M4Wfx+Top587ete2\nufgT6Qekp66pvx3f14dRbTUgNU+g03H9T6N6YNtdkxvgPEKYevmPts4WfSKVj/1Z484UfR6h\nMl0N4eymr8WfSD8gHRudrf2IMDbRh1FtNSA1T6DTcd3PYvLjNz5R3QDncfJ7tTflTU8XfSKP\n/sbLrzw1+sLpos9jtd/8YvyJ9APS7LXPhPDsVWf6MKqtBqTmCXQ6rvdJVG/b2fg364o+j/Dk\nR2qc53/lSNEnsnu00a6izyP83R1nQ5i79nD8ifTlr78fvvn4iVt39WNSWw1I506g03GdOzr2\nxNFarxV9HmHqQ3/40rF7b5ov/ETC6p0p+jwWP/zJo8/97kfL8SfSF0jlR7Zt2b3cj0ltrUBq\nnkCn4zr36MofwF8p+jxCeP4TN3z4/h8VviH1Gnem8PM4+enNH3lwKocTGfh/RYioHwGJKIeA\nRJRDQCLKISAR5RCQiHIISEQ5BCSiHAISUQ4BaZD66taN8R+HG8KANEg9UDpV9CkMa0AapIBU\nWEA6P3r6v/yLf3tjXck3P3Dh2z74d7V3Lruy/utXvjeEHff+/RU/fuGvT4VfLJVKm8P0Xf/m\nTe+8Y6bY8x26gHRe9BdveM8nbv/n7zgdDlxwyZ13XfLGA62Q3v8L7/nNR64t3RSO3lL68rEw\n9s827Rwt3Vj0KQ9ZQDofWn73pbXvMPtLu6rvu+hkCCff9rPVVkil3bX3fu6SlZd2k//ko7WP\nNl9a7BkPXUA6H/p6aU/9sPuJE6Wd9XfuLX2/FdKb6v+DmRt/egXSzD+99IUCT3VYA9L50OdL\nT6+8s6/0WP3w56UDrZB+pv7eTauQwoMXlC777f3Vos51SAPS+dCflI6svLOv9OX64bHS46uQ\nrqhDuqz+3jlI4Ye7r3lL6ZeXCjrXIQ1I50MHS3vrh4/ufql0X/2d3yt9L1z2wfp772yDdOob\nUyEs/LeVb13Ur4B0PjR30chCCH9b+h+V9178eginLrq0Gn7+XeUQHi8JpJPha6X6/7vno6Wv\nFHzOQxaQzou+8GOX/d7db7lkMvzfN/zr/37P2y84EMKO0tjn7rnwPyQg/UHp7oNz777g1x/4\nyE/8O/5lob4GpPOjA7/0kxdf/3LtnW+8/60XfqD+D2QXbrv4J684/MVPtUL6/i+9+bfCd2/4\nl2985y0/LPZ8hy4gEeUQkIhyCEhEOQQkohwCElEOAYkoh4BElENAIsohIBHlEJCIcghIRDkE\nJKIcAhJRDgGJKIf+P1AJOWuZayOlAAAAAElFTkSuQmCC",
      "text/plain": [
       "plot without title"
      ]
     },
     "metadata": {
      "image/png": {
       "height": 420,
       "width": 420
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Some metagenomes have shit depth. Filter those. More specifically, remove those with less than 500 mapped counts\n",
    "dataMetaAdaptedTruncLongAll %>% filter(taxon != \"-1\") %>% group_by(Sample_ID) %>% summarize(counts = sum(count)) %>% ggplot(aes(x = counts)) + geom_histogram(bins = 100)\n",
    "print(dataMetaAdaptedTruncLongAll %>% filter(taxon != \"-1\") %>% group_by(Sample_ID) %>% summarize(counts = sum(count)) %>% filter(counts < rarefyDepth) %>% dim())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "id": "d8916d53-5450-4292-8674-4f4f0c79cf8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` ungrouping output (override with `.groups` argument)\n"
     ]
    }
   ],
   "source": [
    "samplesToKill <- dataMetaAdaptedTruncLongAll %>% \n",
    "filter(taxon != \"-1\") %>%\n",
    "group_by(Sample_ID) %>%\n",
    "summarize(counts = sum(count)) %>%\n",
    "filter(counts < rarefyDepth) %>%\n",
    "pull(Sample_ID)\n",
    "\n",
    "dataMetaAdaptedTruncLongAll <- dataMetaAdaptedTruncLongAll %>%\n",
    "filter(!Sample_ID %in% samplesToKill)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "id": "f5051cee-992a-435a-912b-2f14fca98cca",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "`summarise()` ungrouping output (override with `.groups` argument)\n"
     ]
    }
   ],
   "source": [
    "#TODO: Also check the unmapped fraction over samples to find samples that might be weird?\n",
    "UnassignedFractionInfo <- dataMetaAdaptedTruncLongAll %>%\n",
    "    group_by(Sample_ID) %>%\n",
    "    mutate(relAb = count / sum(count)) %>%\n",
    "    filter(taxon == '-1') %>%\n",
    "    arrange(desc(relAb)) %>%\n",
    "    left_join(dataMetaAdaptedTruncLongAll %>% group_by(Sample_ID) %>% summarize(totalDepth = sum(count)), by = 'Sample_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 422,
   "id": "7251ac9b-e0c5-4071-b7e6-29eb2c0893b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[90m# A tibble: 6,544 x 8\u001b[39m\n",
      "\u001b[90m# Groups:   Sample_ID [6,544]\u001b[39m\n",
      "   taxon Sample_ID    count caseControls dataset      condition relAb totalDepth\n",
      "   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m        \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m        \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m        \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m      \u001b[3m\u001b[90m<dbl>\u001b[39m\u001b[23m\n",
      "\u001b[90m 1\u001b[39m -1    M2061C23_MG   \u001b[4m1\u001b[24m218 control      Lloyd-Price… UC/CD     0.464       \u001b[4m2\u001b[24m627\n",
      "\u001b[90m 2\u001b[39m -1    MetaHIT-MH0…  \u001b[4m9\u001b[24m972 control      LeChatelier… OB        0.431      \u001b[4m2\u001b[24m\u001b[4m3\u001b[24m139\n",
      "\u001b[90m 3\u001b[39m -1    M2061C21_MG   \u001b[4m1\u001b[24m223 control      Lloyd-Price… UC/CD     0.399       \u001b[4m3\u001b[24m062\n",
      "\u001b[90m 4\u001b[39m -1    NG-5636_370   \u001b[4m1\u001b[24m205 case         Karlsson_20… T2D       0.258       \u001b[4m4\u001b[24m678\n",
      "\u001b[90m 5\u001b[39m -1    CCIS3876545…  \u001b[4m4\u001b[24m288 case         Zeller_2014… CRC       0.249      \u001b[4m1\u001b[24m\u001b[4m7\u001b[24m240\n",
      "\u001b[90m 6\u001b[39m -1    713B037-11-…  \u001b[4m4\u001b[24m110 case         Kushugolova… MS        0.248      \u001b[4m1\u001b[24m\u001b[4m6\u001b[24m552\n",
      "\u001b[90m 7\u001b[39m -1    M2061C14_MG    736 control      Lloyd-Price… UC/CD     0.238       \u001b[4m3\u001b[24m086\n",
      "\u001b[90m 8\u001b[39m -1    MetaHIT-MH0…  \u001b[4m5\u001b[24m170 control      LeChatelier… OB        0.223      \u001b[4m2\u001b[24m\u001b[4m3\u001b[24m198\n",
      "\u001b[90m 9\u001b[39m -1    ZSL-146       \u001b[4m3\u001b[24m059 case         Jie_2017_Na… ACVD      0.204      \u001b[4m1\u001b[24m\u001b[4m4\u001b[24m990\n",
      "\u001b[90m10\u001b[39m -1    SAMEA3136660  \u001b[4m1\u001b[24m606 control      Feng_2015_N… ADA/CRC   0.202       \u001b[4m7\u001b[24m948\n",
      "\u001b[90m11\u001b[39m -1    SRR6468600    \u001b[4m1\u001b[24m217 case         Franzosa_Na… UC        0.190       \u001b[4m6\u001b[24m393\n",
      "\u001b[90m12\u001b[39m -1    SRR6468643     617 control      Franzosa_Na… UC/CD     0.184       \u001b[4m3\u001b[24m353\n",
      "\u001b[90m13\u001b[39m -1    C3011C14_MG    555 case         Lloyd-Price… UC        0.184       \u001b[4m3\u001b[24m017\n",
      "\u001b[90m14\u001b[39m -1    MM018          249 control      Vincnet_201… CDI       0.184       \u001b[4m1\u001b[24m354\n",
      "\u001b[90m15\u001b[39m -1    CCIS0626055…   998 case         Zeller_2014… CRC       0.182       \u001b[4m5\u001b[24m480\n",
      "\u001b[90m16\u001b[39m -1    SBJ33C        \u001b[4m1\u001b[24m546 case         Thomas_2019… CRC       0.179       \u001b[4m8\u001b[24m639\n",
      "\u001b[90m17\u001b[39m -1    713B037-11-…  \u001b[4m2\u001b[24m044 case         Kushugolova… MS        0.178      \u001b[4m1\u001b[24m\u001b[4m1\u001b[24m487\n",
      "\u001b[90m18\u001b[39m -1    MetaHIT-MH0…  \u001b[4m2\u001b[24m170 control      LeChatelier… OB        0.178      \u001b[4m1\u001b[24m\u001b[4m2\u001b[24m216\n",
      "\u001b[90m19\u001b[39m -1    SRR6468553     306 case         Franzosa_Na… UC        0.176       \u001b[4m1\u001b[24m741\n",
      "\u001b[90m20\u001b[39m -1    M2061C18_MG    721 control      Lloyd-Price… UC/CD     0.173       \u001b[4m4\u001b[24m178\n",
      "\u001b[90m21\u001b[39m -1    SAMEA3136688  \u001b[4m2\u001b[24m428 case         Feng_2015_N… ADA       0.171      \u001b[4m1\u001b[24m\u001b[4m4\u001b[24m182\n",
      "\u001b[90m22\u001b[39m -1    M2061C10_MG    492 control      Lloyd-Price… UC/CD     0.169       \u001b[4m2\u001b[24m911\n",
      "\u001b[90m23\u001b[39m -1    M2061C16_MG    463 control      Lloyd-Price… UC/CD     0.166       \u001b[4m2\u001b[24m785\n",
      "\u001b[90m24\u001b[39m -1    M2061C17_MG    428 control      Lloyd-Price… UC/CD     0.162       \u001b[4m2\u001b[24m646\n",
      "\u001b[90m25\u001b[39m -1    MetaHIT-MH0…  \u001b[4m4\u001b[24m206 control      LeChatelier… OB        0.160      \u001b[4m2\u001b[24m\u001b[4m6\u001b[24m336\n",
      "\u001b[90m26\u001b[39m -1    SRR6468682    \u001b[4m1\u001b[24m291 case         Franzosa_Na… CD        0.159       \u001b[4m8\u001b[24m118\n",
      "\u001b[90m27\u001b[39m -1    713B040-11-…  \u001b[4m3\u001b[24m560 case         Kushugolova… MS        0.153      \u001b[4m2\u001b[24m\u001b[4m3\u001b[24m233\n",
      "\u001b[90m28\u001b[39m -1    CCMD7934950…   826 case         Wirbel_2019… CRC       0.151       \u001b[4m5\u001b[24m457\n",
      "\u001b[90m29\u001b[39m -1    C3011C10_MG    280 case         Lloyd-Price… UC        0.151       \u001b[4m1\u001b[24m852\n",
      "\u001b[90m30\u001b[39m -1    SRR6468550     428 control      Franzosa_Na… UC/CD     0.151       \u001b[4m2\u001b[24m833\n",
      "\u001b[90m# … with 6,514 more rows\u001b[39m\n"
     ]
    }
   ],
   "source": [
    "UnassignedFractionInfo %>% print(n=30)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "id": "272ce0f5-6cae-4a5f-b610-6ca7c262f7f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Get Rarefied counts for fair diversity score calculations\n",
    "getRarefiedCountTable <- function(countTable, rarefyDepth) {\n",
    "    set.seed(1)\n",
    "    countTableRarefied <- countTable %>% \n",
    "    # I explicitly resample using the unmapped fraction. It's going to be gone later, anyway :)\n",
    "    left_joinate(taxon = as.factor(taxon)) %>%\n",
    "    group_by(Sample_ID) %>%\n",
    "    nest() %>% \n",
    "   #mutate(N = map_dbl(data, function(x) dim(x)[1] %>% return())) %>%\n",
    "    mutate(data = map(data, function(x) {\n",
    "        x <- x %>% arrange(taxon)\n",
    "        tmp <- table(factor(sample(x = x$taxon, size = rarefyDepth, prob = x$count, replace = T), levels = levels(x$taxon)))\n",
    "        stopifnot(all(names(tmp) == x$taxon))\n",
    "        x$count_resampled <- as.vector(tmp)\n",
    "        return(x %>%\n",
    "              select(-count))\n",
    "    })) %>%\n",
    "    unnest()\n",
    "   \n",
    "    return(countTableRarefied)\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "id": "1a5bd032-4d55-4ac6-ac00-112202a110c9",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message:\n",
      "“`cols` is now required when using unnest().\n",
      "Please use `cols = c(data)`”\n"
     ]
    }
   ],
   "source": [
    "dataMetaAdaptedTruncLongAllRarefied <- getRarefiedCountTable(dataMetaAdaptedTruncLongAll, rarefyDepth)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 415,
   "id": "330d63bc-fce5-4d7f-9e22-56ad27d00bf8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get relative abundances\n",
    "dataMetaAdaptedTruncLongAll <- dataMetaAdaptedTruncLongAll %>%\n",
    "group_by(Sample_ID) %>%\n",
    "mutate(relAb = count / sum(count)) %>%\n",
    "# Remove unmapped fraction\n",
    "filter(taxon != \"-1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 416,
   "id": "546cfb92-3c15-41eb-aedf-317cc2fdfd47",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataMetaAdaptedTruncLongAllRarefied <- dataMetaAdaptedTruncLongAllRarefied %>%\n",
    "group_by(Sample_ID) %>%\n",
    "mutate(relAb = count_resampled / sum(count_resampled)) %>%\n",
    "# Remove unmapped fraction\n",
    "filter(taxon != \"-1\")\n",
    "\n",
    "# Check that each sample contains the union of all taxa... then pivot_wider.\n",
    "stopifnot(length(unique(dataMetaAdaptedTruncLongAll %>% group_by(Sample_ID) %>% tally() %>% pull(n))) == 1)\n",
    "\n",
    "dataMetaAdaptedTruncWideAll <- dataMetaAdaptedTruncLongAll %>%\n",
    "pivot_wider(id_cols = taxon, names_from = Sample_ID, values_from = relAb)\n",
    "\n",
    "stopifnot(length(unique(dataMetaAdaptedTruncLongAllRarefied %>% group_by(Sample_ID) %>% tally() %>% pull(n))) == 1)\n",
    "\n",
    "dataMetaAdaptedTruncWideAllRarefied <- dataMetaAdaptedTruncLongAllRarefied %>%\n",
    "mutate(taxon = as.character(taxon)) %>%\n",
    "filter(taxon != '-1') %>%\n",
    "pivot_wider(id_cols = taxon, names_from = Sample_ID, values_from = relAb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "id": "965cf985-4003-4abf-9b46-a10d7d833ffb",
   "metadata": {},
   "outputs": [],
   "source": [
    "metaDataWGS <- dataMetaAdaptedTruncLongAll %>% \n",
    "ungroup() %>%\n",
    "select(Sample_ID, caseControls, dataset, condition) %>% \n",
    "distinct() %>%\n",
    "mutate(caseControls = str_replace(string = caseControls, \n",
    "                                  pattern = \"CTR\", \n",
    "                                  \"control\")) %>%\n",
    "                      as.data.frame() %>%\n",
    "                      mutate(caseControls = as.factor(caseControls))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "id": "26e39bfb-ee58-470b-8d9f-9d335b4201d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataMetaAdaptedTruncWideAll <- dataMetaAdaptedTruncWideAll %>%\n",
    "as.data.frame()\n",
    "rownames(dataMetaAdaptedTruncWideAll) <- dataMetaAdaptedTruncWideAll$taxon\n",
    "dataMetaAdaptedTruncWideAll$taxon <- NULL\n",
    "\n",
    "dataMetaAdaptedTruncWideAllRarefied <- dataMetaAdaptedTruncWideAllRarefied %>%\n",
    "as.data.frame()\n",
    "rownames(dataMetaAdaptedTruncWideAllRarefied) <- dataMetaAdaptedTruncWideAllRarefied$taxon\n",
    "dataMetaAdaptedTruncWideAllRarefied$taxon <- NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "id": "9010d85c-8070-4209-8b0b-10726292ee68",
   "metadata": {},
   "outputs": [],
   "source": [
    "rownames(metaDataWGS) <- metaDataWGS$Sample_ID\n",
    "metaDataWGS$Sample_ID <- NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 420,
   "id": "20115fa5-255f-499c-9c3e-add78f7320c0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Calculcate diversity estimates. Later on join onto metaDataWGS.\n",
    "# TODO: Check if richness == richnessWithCutoff. I think it is due to subsampling.\n",
    "allDat <- getFeatures(dataMetaAdaptedTruncWideAllRarefied, metaDataWGS) \n",
    "morePredictorNames <- allDat[[2]]\n",
    "allDat <- allDat[[1]]    \n",
    "nn <- names(allDat)                           \n",
    "stopifnot(all(map_dbl(allDat, length) == dim(dataMetaAdaptedTruncWideAll)[2]))    \n",
    "\n",
    "tmpData <- matrix(unlist(allDat), byrow = T, nrow = length(allDat)) %>%\n",
    "                                 as.data.frame()\n",
    "colnames(tmpData) <- colnames(dataMetaAdaptedTruncWideAll)\n",
    "rownames(tmpData) <- nn   \n",
    "tmpData <- t(tmpData)\n",
    "stopifnot(all(rownames(tmpData) == rownames(metaDataWGS)))\n",
    "metaDataWGS <- metaDataWGS %>% \n",
    "                                 as.data.frame() %>% \n",
    "                                 mutate(sampleID = rownames(.)) %>% \n",
    "                                 left_join(tmpData %>% \n",
    "                                           as.data.frame() %>% \n",
    "                                           mutate(sampleID = rownames(.)), \n",
    "                                           by ='sampleID')                                \n",
    "rownames(metaDataWGS) <- metaDataWGS$sampleID\n",
    "metaDataWGS$sampleID <- NULL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "id": "6ec1d1ad-07ac-48be-a78b-e8688cd37611",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Parsed with column specification:\n",
      "cols(\n",
      "  mOTUs_ID = \u001b[31mcol_character()\u001b[39m,\n",
      "  Kingdom = \u001b[31mcol_character()\u001b[39m,\n",
      "  Phylum = \u001b[31mcol_character()\u001b[39m,\n",
      "  Class = \u001b[31mcol_character()\u001b[39m,\n",
      "  Order = \u001b[31mcol_character()\u001b[39m,\n",
      "  Family = \u001b[31mcol_character()\u001b[39m,\n",
      "  Genus = \u001b[31mcol_character()\u001b[39m,\n",
      "  Species = \u001b[31mcol_character()\u001b[39m,\n",
      "  profiled = \u001b[31mcol_character()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>mOTUs_ID</th><th scope=col>Kingdom</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th><th scope=col>profiled</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>ref_mOTU_v25_11388</td><td>2 Bacteria</td><td>1239 Firmicutes      </td><td>186801 Clostridia        </td><td>68295 Thermoanaerobacterales</td><td>543371 Thermoanaerobacterales Family III. Incertae Sedis</td><td>159730 Thermovenabulum</td><td>520767 Thermovenabulum gondwanense</td><td>520767 Thermovenabulum gondwanense [ref_mOTU_v25_11388]</td></tr>\n",
       "\t<tr><td>ref_mOTU_v25_02118</td><td>2 Bacteria</td><td>1239 Firmicutes      </td><td>186801 Clostridia        </td><td>186802 Clostridiales        </td><td>1491775 Proteinivoraceae                                </td><td>42447 Anaerobranca    </td><td>108328 Anaerobranca gottschalkii  </td><td>108328 Anaerobranca gottschalkii [ref_mOTU_v25_02118]  </td></tr>\n",
       "\t<tr><td>ref_mOTU_v25_12184</td><td>2 Bacteria</td><td>976 Bacteroidetes    </td><td>768503 Cytophagia        </td><td>768507 Cytophagales         </td><td>NA Cytophagales fam. incertae sedis                     </td><td>1433993 Chryseolinea  </td><td>947013 Chryseolinea serpens       </td><td>947013 Chryseolinea serpens [ref_mOTU_v25_12184]       </td></tr>\n",
       "\t<tr><td>ref_mOTU_v25_06844</td><td>2 Bacteria</td><td>1239 Firmicutes      </td><td>186801 Clostridia        </td><td>186802 Clostridiales        </td><td>186807 Peptococcaceae                                   </td><td>1562 Desulfotomaculum </td><td>74701 Desulfotomaculum putei      </td><td>74701 Desulfotomaculum putei [ref_mOTU_v25_06844]      </td></tr>\n",
       "\t<tr><td>ref_mOTU_v25_05295</td><td>2 Bacteria</td><td>201174 Actinobacteria</td><td>1760 Actinobacteria      </td><td>85006 Micrococcales         </td><td>145360 Sanguibacteraceae                                </td><td>60919 Sanguibacter    </td><td>60920 Sanguibacter keddieii       </td><td>60920 Sanguibacter keddieii [ref_mOTU_v25_05295]       </td></tr>\n",
       "\t<tr><td>ref_mOTU_v25_01391</td><td>2 Bacteria</td><td>1224 Proteobacteria  </td><td>28211 Alphaproteobacteria</td><td>356 Rhizobiales             </td><td>82115 Rhizobiaceae                                      </td><td>357 Agrobacterium     </td><td>358 Agrobacterium tumefaciens     </td><td>358 Agrobacterium tumefaciens [ref_mOTU_v25_01391]     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 9\n",
       "\\begin{tabular}{lllllllll}\n",
       " mOTUs\\_ID & Kingdom & Phylum & Class & Order & Family & Genus & Species & profiled\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t ref\\_mOTU\\_v25\\_11388 & 2 Bacteria & 1239 Firmicutes       & 186801 Clostridia         & 68295 Thermoanaerobacterales & 543371 Thermoanaerobacterales Family III. Incertae Sedis & 159730 Thermovenabulum & 520767 Thermovenabulum gondwanense & 520767 Thermovenabulum gondwanense {[}ref\\_mOTU\\_v25\\_11388{]}\\\\\n",
       "\t ref\\_mOTU\\_v25\\_02118 & 2 Bacteria & 1239 Firmicutes       & 186801 Clostridia         & 186802 Clostridiales         & 1491775 Proteinivoraceae                                 & 42447 Anaerobranca     & 108328 Anaerobranca gottschalkii   & 108328 Anaerobranca gottschalkii {[}ref\\_mOTU\\_v25\\_02118{]}  \\\\\n",
       "\t ref\\_mOTU\\_v25\\_12184 & 2 Bacteria & 976 Bacteroidetes     & 768503 Cytophagia         & 768507 Cytophagales          & NA Cytophagales fam. incertae sedis                      & 1433993 Chryseolinea   & 947013 Chryseolinea serpens        & 947013 Chryseolinea serpens {[}ref\\_mOTU\\_v25\\_12184{]}       \\\\\n",
       "\t ref\\_mOTU\\_v25\\_06844 & 2 Bacteria & 1239 Firmicutes       & 186801 Clostridia         & 186802 Clostridiales         & 186807 Peptococcaceae                                    & 1562 Desulfotomaculum  & 74701 Desulfotomaculum putei       & 74701 Desulfotomaculum putei {[}ref\\_mOTU\\_v25\\_06844{]}      \\\\\n",
       "\t ref\\_mOTU\\_v25\\_05295 & 2 Bacteria & 201174 Actinobacteria & 1760 Actinobacteria       & 85006 Micrococcales          & 145360 Sanguibacteraceae                                 & 60919 Sanguibacter     & 60920 Sanguibacter keddieii        & 60920 Sanguibacter keddieii {[}ref\\_mOTU\\_v25\\_05295{]}       \\\\\n",
       "\t ref\\_mOTU\\_v25\\_01391 & 2 Bacteria & 1224 Proteobacteria   & 28211 Alphaproteobacteria & 356 Rhizobiales              & 82115 Rhizobiaceae                                       & 357 Agrobacterium      & 358 Agrobacterium tumefaciens      & 358 Agrobacterium tumefaciens {[}ref\\_mOTU\\_v25\\_01391{]}     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 9\n",
       "\n",
       "| mOTUs_ID &lt;chr&gt; | Kingdom &lt;chr&gt; | Phylum &lt;chr&gt; | Class &lt;chr&gt; | Order &lt;chr&gt; | Family &lt;chr&gt; | Genus &lt;chr&gt; | Species &lt;chr&gt; | profiled &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| ref_mOTU_v25_11388 | 2 Bacteria | 1239 Firmicutes       | 186801 Clostridia         | 68295 Thermoanaerobacterales | 543371 Thermoanaerobacterales Family III. Incertae Sedis | 159730 Thermovenabulum | 520767 Thermovenabulum gondwanense | 520767 Thermovenabulum gondwanense [ref_mOTU_v25_11388] |\n",
       "| ref_mOTU_v25_02118 | 2 Bacteria | 1239 Firmicutes       | 186801 Clostridia         | 186802 Clostridiales         | 1491775 Proteinivoraceae                                 | 42447 Anaerobranca     | 108328 Anaerobranca gottschalkii   | 108328 Anaerobranca gottschalkii [ref_mOTU_v25_02118]   |\n",
       "| ref_mOTU_v25_12184 | 2 Bacteria | 976 Bacteroidetes     | 768503 Cytophagia         | 768507 Cytophagales          | NA Cytophagales fam. incertae sedis                      | 1433993 Chryseolinea   | 947013 Chryseolinea serpens        | 947013 Chryseolinea serpens [ref_mOTU_v25_12184]        |\n",
       "| ref_mOTU_v25_06844 | 2 Bacteria | 1239 Firmicutes       | 186801 Clostridia         | 186802 Clostridiales         | 186807 Peptococcaceae                                    | 1562 Desulfotomaculum  | 74701 Desulfotomaculum putei       | 74701 Desulfotomaculum putei [ref_mOTU_v25_06844]       |\n",
       "| ref_mOTU_v25_05295 | 2 Bacteria | 201174 Actinobacteria | 1760 Actinobacteria       | 85006 Micrococcales          | 145360 Sanguibacteraceae                                 | 60919 Sanguibacter     | 60920 Sanguibacter keddieii        | 60920 Sanguibacter keddieii [ref_mOTU_v25_05295]        |\n",
       "| ref_mOTU_v25_01391 | 2 Bacteria | 1224 Proteobacteria   | 28211 Alphaproteobacteria | 356 Rhizobiales              | 82115 Rhizobiaceae                                       | 357 Agrobacterium      | 358 Agrobacterium tumefaciens      | 358 Agrobacterium tumefaciens [ref_mOTU_v25_01391]      |\n",
       "\n"
      ],
      "text/plain": [
       "  mOTUs_ID           Kingdom    Phylum                Class                    \n",
       "1 ref_mOTU_v25_11388 2 Bacteria 1239 Firmicutes       186801 Clostridia        \n",
       "2 ref_mOTU_v25_02118 2 Bacteria 1239 Firmicutes       186801 Clostridia        \n",
       "3 ref_mOTU_v25_12184 2 Bacteria 976 Bacteroidetes     768503 Cytophagia        \n",
       "4 ref_mOTU_v25_06844 2 Bacteria 1239 Firmicutes       186801 Clostridia        \n",
       "5 ref_mOTU_v25_05295 2 Bacteria 201174 Actinobacteria 1760 Actinobacteria      \n",
       "6 ref_mOTU_v25_01391 2 Bacteria 1224 Proteobacteria   28211 Alphaproteobacteria\n",
       "  Order                       \n",
       "1 68295 Thermoanaerobacterales\n",
       "2 186802 Clostridiales        \n",
       "3 768507 Cytophagales         \n",
       "4 186802 Clostridiales        \n",
       "5 85006 Micrococcales         \n",
       "6 356 Rhizobiales             \n",
       "  Family                                                  \n",
       "1 543371 Thermoanaerobacterales Family III. Incertae Sedis\n",
       "2 1491775 Proteinivoraceae                                \n",
       "3 NA Cytophagales fam. incertae sedis                     \n",
       "4 186807 Peptococcaceae                                   \n",
       "5 145360 Sanguibacteraceae                                \n",
       "6 82115 Rhizobiaceae                                      \n",
       "  Genus                  Species                           \n",
       "1 159730 Thermovenabulum 520767 Thermovenabulum gondwanense\n",
       "2 42447 Anaerobranca     108328 Anaerobranca gottschalkii  \n",
       "3 1433993 Chryseolinea   947013 Chryseolinea serpens       \n",
       "4 1562 Desulfotomaculum  74701 Desulfotomaculum putei      \n",
       "5 60919 Sanguibacter     60920 Sanguibacter keddieii       \n",
       "6 357 Agrobacterium      358 Agrobacterium tumefaciens     \n",
       "  profiled                                               \n",
       "1 520767 Thermovenabulum gondwanense [ref_mOTU_v25_11388]\n",
       "2 108328 Anaerobranca gottschalkii [ref_mOTU_v25_02118]  \n",
       "3 947013 Chryseolinea serpens [ref_mOTU_v25_12184]       \n",
       "4 74701 Desulfotomaculum putei [ref_mOTU_v25_06844]      \n",
       "5 60920 Sanguibacter keddieii [ref_mOTU_v25_05295]       \n",
       "6 358 Agrobacterium tumefaciens [ref_mOTU_v25_01391]     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "read_tsv('../data/motus2.6_taxonomy_NCBI.tax') %>% head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "3cbde66a-e9ac-4513-9577-72128d16fa2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6497 × 15</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>caseControls</th><th scope=col>dataset</th><th scope=col>condition</th><th scope=col>richness</th><th scope=col>richnessWithCutoff</th><th scope=col>richnessAbsentInHC</th><th scope=col>richnessWithCutoffAbsentInHCWithCutoff</th><th scope=col>cumAbundanceAbsentInHC</th><th scope=col>cumAbundanceAbsentInHCWithCutoff</th><th scope=col>richnessPresentInHC</th><th scope=col>richnessWithCutoffPresentInHCWithCutoff</th><th scope=col>cumAbundancePresentInHC</th><th scope=col>cumAbundancePresentInHCWithCutoff</th><th scope=col>topTaxonRelAb</th><th scope=col>topTaxaRelAb</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>SAMEA3136748</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td> 87</td><td> 87</td><td> 8</td><td> 8</td><td>0.022</td><td>0.022</td><td>18</td><td>18</td><td>0.350</td><td>0.350</td><td>0.064</td><td>0.013666667</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136648</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>100</td><td>100</td><td>27</td><td>27</td><td>0.260</td><td>0.260</td><td>13</td><td>13</td><td>0.190</td><td>0.190</td><td>0.110</td><td>0.017000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136638</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 99</td><td> 99</td><td>14</td><td>14</td><td>0.040</td><td>0.040</td><td>18</td><td>18</td><td>0.382</td><td>0.382</td><td>0.078</td><td>0.003666667</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136670</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 93</td><td> 93</td><td>15</td><td>15</td><td>0.058</td><td>0.058</td><td>14</td><td>14</td><td>0.272</td><td>0.272</td><td>0.086</td><td>0.004666667</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136657</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 49</td><td> 49</td><td>16</td><td>16</td><td>0.162</td><td>0.162</td><td> 7</td><td> 7</td><td>0.276</td><td>0.276</td><td>0.150</td><td>0.006500000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136660</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 75</td><td> 75</td><td>17</td><td>17</td><td>0.050</td><td>0.050</td><td>10</td><td>10</td><td>0.340</td><td>0.340</td><td>0.208</td><td>0.008800000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136769</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td> 49</td><td> 49</td><td>10</td><td>10</td><td>0.064</td><td>0.064</td><td> 6</td><td> 6</td><td>0.156</td><td>0.156</td><td>0.204</td><td>0.005000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136630</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>107</td><td>107</td><td>27</td><td>27</td><td>0.164</td><td>0.164</td><td>14</td><td>14</td><td>0.360</td><td>0.360</td><td>0.068</td><td>0.003714286</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136751</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td>124</td><td>124</td><td>25</td><td>25</td><td>0.118</td><td>0.118</td><td>13</td><td>13</td><td>0.142</td><td>0.142</td><td>0.122</td><td>0.003000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136634</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>113</td><td>113</td><td>16</td><td>16</td><td>0.052</td><td>0.052</td><td>16</td><td>16</td><td>0.296</td><td>0.296</td><td>0.090</td><td>0.010571429</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136738</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td>119</td><td>119</td><td>54</td><td>54</td><td>0.524</td><td>0.524</td><td> 5</td><td> 5</td><td>0.016</td><td>0.016</td><td>0.070</td><td>0.003714286</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136674</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 64</td><td> 64</td><td> 6</td><td> 6</td><td>0.024</td><td>0.024</td><td>17</td><td>17</td><td>0.548</td><td>0.548</td><td>0.112</td><td>0.026000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136663</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 80</td><td> 80</td><td>10</td><td>10</td><td>0.058</td><td>0.058</td><td>18</td><td>18</td><td>0.522</td><td>0.522</td><td>0.266</td><td>0.055200000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136747</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td> 62</td><td> 62</td><td> 9</td><td> 9</td><td>0.068</td><td>0.068</td><td>10</td><td>10</td><td>0.202</td><td>0.202</td><td>0.172</td><td>0.008000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136676</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>113</td><td>113</td><td>22</td><td>22</td><td>0.134</td><td>0.134</td><td>18</td><td>18</td><td>0.348</td><td>0.348</td><td>0.160</td><td>0.027428571</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136743</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td>118</td><td>118</td><td>19</td><td>19</td><td>0.116</td><td>0.116</td><td>13</td><td>13</td><td>0.220</td><td>0.220</td><td>0.112</td><td>0.020285714</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136754</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td>118</td><td>118</td><td>20</td><td>20</td><td>0.170</td><td>0.170</td><td>18</td><td>18</td><td>0.218</td><td>0.218</td><td>0.086</td><td>0.014857143</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136644</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>141</td><td>141</td><td>32</td><td>32</td><td>0.260</td><td>0.260</td><td>20</td><td>20</td><td>0.194</td><td>0.194</td><td>0.070</td><td>0.008888889</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136654</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td>128</td><td>128</td><td>24</td><td>24</td><td>0.148</td><td>0.148</td><td>21</td><td>21</td><td>0.240</td><td>0.240</td><td>0.060</td><td>0.007250000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136673</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 88</td><td> 88</td><td>12</td><td>12</td><td>0.114</td><td>0.114</td><td>17</td><td>17</td><td>0.448</td><td>0.448</td><td>0.124</td><td>0.023666667</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136665</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 57</td><td> 57</td><td>15</td><td>15</td><td>0.312</td><td>0.312</td><td>16</td><td>16</td><td>0.160</td><td>0.160</td><td>0.256</td><td>0.011000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136664</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 88</td><td> 88</td><td> 5</td><td> 5</td><td>0.026</td><td>0.026</td><td>21</td><td>21</td><td>0.288</td><td>0.288</td><td>0.158</td><td>0.004333333</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136765</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td> 83</td><td> 83</td><td>20</td><td>20</td><td>0.072</td><td>0.072</td><td>13</td><td>13</td><td>0.198</td><td>0.198</td><td>0.470</td><td>0.008666667</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136656</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 70</td><td> 70</td><td>10</td><td>10</td><td>0.024</td><td>0.024</td><td>13</td><td>13</td><td>0.408</td><td>0.408</td><td>0.180</td><td>0.007200000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136650</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 40</td><td> 40</td><td>17</td><td>17</td><td>0.158</td><td>0.158</td><td> 4</td><td> 4</td><td>0.666</td><td>0.666</td><td>0.540</td><td>0.004000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136626</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 48</td><td> 48</td><td> 8</td><td> 8</td><td>0.046</td><td>0.046</td><td> 9</td><td> 9</td><td>0.672</td><td>0.672</td><td>0.264</td><td>0.019000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136724</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td> 70</td><td> 70</td><td> 7</td><td> 7</td><td>0.022</td><td>0.022</td><td>16</td><td>16</td><td>0.378</td><td>0.378</td><td>0.128</td><td>0.005200000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136668</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 58</td><td> 58</td><td>15</td><td>15</td><td>0.074</td><td>0.074</td><td> 7</td><td> 7</td><td>0.308</td><td>0.308</td><td>0.222</td><td>0.003000000</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136645</th><td>control</td><td>Feng_2015_NatComm</td><td>CRC</td><td> 89</td><td> 89</td><td>15</td><td>15</td><td>0.064</td><td>0.064</td><td>14</td><td>14</td><td>0.210</td><td>0.210</td><td>0.264</td><td>0.010333333</td></tr>\n",
       "\t<tr><th scope=row>SAMEA3136755</th><td>case   </td><td>Feng_2015_NatComm</td><td>CRC</td><td>109</td><td>109</td><td>30</td><td>30</td><td>0.128</td><td>0.128</td><td>15</td><td>15</td><td>0.300</td><td>0.300</td><td>0.152</td><td>0.002571429</td></tr>\n",
       "\t<tr><th scope=row>⋮</th><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td><td>⋮</td></tr>\n",
       "\t<tr><th scope=row>WGC076504</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 52</td><td> 52</td><td> 7</td><td> 7</td><td>0.052</td><td>0.052</td><td>10</td><td>10</td><td>0.420</td><td>0.420</td><td>0.176</td><td>0.010500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076505</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td>108</td><td>108</td><td>13</td><td>13</td><td>0.094</td><td>0.094</td><td>19</td><td>19</td><td>0.254</td><td>0.254</td><td>0.174</td><td>0.006285714</td></tr>\n",
       "\t<tr><th scope=row>WGC076506</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 68</td><td> 68</td><td>12</td><td>12</td><td>0.070</td><td>0.070</td><td>16</td><td>16</td><td>0.540</td><td>0.540</td><td>0.228</td><td>0.003600000</td></tr>\n",
       "\t<tr><th scope=row>WGC076507</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 76</td><td> 76</td><td> 5</td><td> 5</td><td>0.068</td><td>0.068</td><td>16</td><td>16</td><td>0.124</td><td>0.124</td><td>0.256</td><td>0.006000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076508</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td>109</td><td>109</td><td>11</td><td>11</td><td>0.024</td><td>0.024</td><td>20</td><td>20</td><td>0.192</td><td>0.192</td><td>0.254</td><td>0.007142857</td></tr>\n",
       "\t<tr><th scope=row>WGC076509</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 86</td><td> 86</td><td>10</td><td>10</td><td>0.062</td><td>0.062</td><td>21</td><td>21</td><td>0.194</td><td>0.194</td><td>0.250</td><td>0.004333333</td></tr>\n",
       "\t<tr><th scope=row>WGC076510</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 69</td><td> 69</td><td> 4</td><td> 4</td><td>0.014</td><td>0.014</td><td>18</td><td>18</td><td>0.528</td><td>0.528</td><td>0.194</td><td>0.004400000</td></tr>\n",
       "\t<tr><th scope=row>WGC076511</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td>111</td><td>111</td><td> 9</td><td> 9</td><td>0.018</td><td>0.018</td><td>20</td><td>20</td><td>0.418</td><td>0.418</td><td>0.092</td><td>0.003714286</td></tr>\n",
       "\t<tr><th scope=row>WGC076513</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 39</td><td> 39</td><td> 3</td><td> 3</td><td>0.006</td><td>0.006</td><td> 9</td><td> 9</td><td>0.776</td><td>0.776</td><td>0.424</td><td>0.026666667</td></tr>\n",
       "\t<tr><th scope=row>WGC076515</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 67</td><td> 67</td><td> 3</td><td> 3</td><td>0.010</td><td>0.010</td><td>17</td><td>17</td><td>0.372</td><td>0.372</td><td>0.222</td><td>0.010400000</td></tr>\n",
       "\t<tr><th scope=row>WGC076516</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 81</td><td> 81</td><td> 6</td><td> 6</td><td>0.034</td><td>0.034</td><td>16</td><td>16</td><td>0.340</td><td>0.340</td><td>0.118</td><td>0.006000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076517</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 59</td><td> 59</td><td> 4</td><td> 4</td><td>0.060</td><td>0.060</td><td>13</td><td>13</td><td>0.466</td><td>0.466</td><td>0.212</td><td>0.007000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076518</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 30</td><td> 30</td><td> 2</td><td> 2</td><td>0.008</td><td>0.008</td><td>14</td><td>14</td><td>0.614</td><td>0.614</td><td>0.484</td><td>0.006000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076520</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 88</td><td> 88</td><td>18</td><td>18</td><td>0.064</td><td>0.064</td><td>17</td><td>17</td><td>0.410</td><td>0.410</td><td>0.184</td><td>0.003000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076521</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 55</td><td> 55</td><td>15</td><td>15</td><td>0.058</td><td>0.058</td><td> 6</td><td> 6</td><td>0.466</td><td>0.466</td><td>0.246</td><td>0.002500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076522</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 41</td><td> 41</td><td> 4</td><td> 4</td><td>0.570</td><td>0.570</td><td>15</td><td>15</td><td>0.152</td><td>0.152</td><td>0.306</td><td>0.013500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076523</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 60</td><td> 60</td><td> 5</td><td> 5</td><td>0.026</td><td>0.026</td><td>14</td><td>14</td><td>0.622</td><td>0.622</td><td>0.400</td><td>0.006500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076524</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 73</td><td> 73</td><td> 4</td><td> 4</td><td>0.022</td><td>0.022</td><td>21</td><td>21</td><td>0.362</td><td>0.362</td><td>0.200</td><td>0.004000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076525</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 41</td><td> 41</td><td> 4</td><td> 4</td><td>0.256</td><td>0.256</td><td> 9</td><td> 9</td><td>0.078</td><td>0.078</td><td>0.460</td><td>0.063500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076526</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 51</td><td> 51</td><td> 4</td><td> 4</td><td>0.008</td><td>0.008</td><td>12</td><td>12</td><td>0.672</td><td>0.672</td><td>0.218</td><td>0.002000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076527</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 65</td><td> 65</td><td> 4</td><td> 4</td><td>0.008</td><td>0.008</td><td>16</td><td>16</td><td>0.312</td><td>0.312</td><td>0.184</td><td>0.003600000</td></tr>\n",
       "\t<tr><th scope=row>WGC076528</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td>102</td><td>102</td><td>10</td><td>10</td><td>0.068</td><td>0.068</td><td>19</td><td>19</td><td>0.366</td><td>0.366</td><td>0.138</td><td>0.007428571</td></tr>\n",
       "\t<tr><th scope=row>WGC076529</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 27</td><td> 27</td><td> 9</td><td> 9</td><td>0.078</td><td>0.078</td><td> 4</td><td> 4</td><td>0.646</td><td>0.646</td><td>0.348</td><td>0.013333333</td></tr>\n",
       "\t<tr><th scope=row>WGC076531</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 72</td><td> 72</td><td> 6</td><td> 6</td><td>0.136</td><td>0.136</td><td>16</td><td>16</td><td>0.394</td><td>0.394</td><td>0.158</td><td>0.006400000</td></tr>\n",
       "\t<tr><th scope=row>WGC076532</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 36</td><td> 36</td><td> 7</td><td> 7</td><td>0.060</td><td>0.060</td><td>10</td><td>10</td><td>0.184</td><td>0.184</td><td>0.502</td><td>0.014666667</td></tr>\n",
       "\t<tr><th scope=row>WGC076533</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 51</td><td> 51</td><td> 6</td><td> 6</td><td>0.014</td><td>0.014</td><td>15</td><td>15</td><td>0.468</td><td>0.468</td><td>0.332</td><td>0.005500000</td></tr>\n",
       "\t<tr><th scope=row>WGC076534</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 36</td><td> 36</td><td> 3</td><td> 3</td><td>0.012</td><td>0.012</td><td>10</td><td>10</td><td>0.306</td><td>0.306</td><td>0.452</td><td>0.003333333</td></tr>\n",
       "\t<tr><th scope=row>WGC076537</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 53</td><td> 53</td><td> 9</td><td> 9</td><td>0.036</td><td>0.036</td><td>11</td><td>11</td><td>0.708</td><td>0.708</td><td>0.510</td><td>0.004000000</td></tr>\n",
       "\t<tr><th scope=row>WGC076538</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 68</td><td> 68</td><td> 7</td><td> 7</td><td>0.014</td><td>0.014</td><td>16</td><td>16</td><td>0.544</td><td>0.544</td><td>0.290</td><td>0.005200000</td></tr>\n",
       "\t<tr><th scope=row>WGC076539</th><td>control</td><td>Zhu_2018_Microbiome</td><td>BC</td><td> 79</td><td> 79</td><td> 3</td><td> 3</td><td>0.006</td><td>0.006</td><td>19</td><td>19</td><td>0.586</td><td>0.586</td><td>0.234</td><td>0.007200000</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6497 × 15\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & caseControls & dataset & condition & richness & richnessWithCutoff & richnessAbsentInHC & richnessWithCutoffAbsentInHCWithCutoff & cumAbundanceAbsentInHC & cumAbundanceAbsentInHCWithCutoff & richnessPresentInHC & richnessWithCutoffPresentInHCWithCutoff & cumAbundancePresentInHC & cumAbundancePresentInHCWithCutoff & topTaxonRelAb & topTaxaRelAb\\\\\n",
       "  & <fct> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\tSAMEA3136748 & case    & Feng\\_2015\\_NatComm & CRC &  87 &  87 &  8 &  8 & 0.022 & 0.022 & 18 & 18 & 0.350 & 0.350 & 0.064 & 0.013666667\\\\\n",
       "\tSAMEA3136648 & control & Feng\\_2015\\_NatComm & CRC & 100 & 100 & 27 & 27 & 0.260 & 0.260 & 13 & 13 & 0.190 & 0.190 & 0.110 & 0.017000000\\\\\n",
       "\tSAMEA3136638 & control & Feng\\_2015\\_NatComm & CRC &  99 &  99 & 14 & 14 & 0.040 & 0.040 & 18 & 18 & 0.382 & 0.382 & 0.078 & 0.003666667\\\\\n",
       "\tSAMEA3136670 & control & Feng\\_2015\\_NatComm & CRC &  93 &  93 & 15 & 15 & 0.058 & 0.058 & 14 & 14 & 0.272 & 0.272 & 0.086 & 0.004666667\\\\\n",
       "\tSAMEA3136657 & control & Feng\\_2015\\_NatComm & CRC &  49 &  49 & 16 & 16 & 0.162 & 0.162 &  7 &  7 & 0.276 & 0.276 & 0.150 & 0.006500000\\\\\n",
       "\tSAMEA3136660 & control & Feng\\_2015\\_NatComm & CRC &  75 &  75 & 17 & 17 & 0.050 & 0.050 & 10 & 10 & 0.340 & 0.340 & 0.208 & 0.008800000\\\\\n",
       "\tSAMEA3136769 & case    & Feng\\_2015\\_NatComm & CRC &  49 &  49 & 10 & 10 & 0.064 & 0.064 &  6 &  6 & 0.156 & 0.156 & 0.204 & 0.005000000\\\\\n",
       "\tSAMEA3136630 & control & Feng\\_2015\\_NatComm & CRC & 107 & 107 & 27 & 27 & 0.164 & 0.164 & 14 & 14 & 0.360 & 0.360 & 0.068 & 0.003714286\\\\\n",
       "\tSAMEA3136751 & case    & Feng\\_2015\\_NatComm & CRC & 124 & 124 & 25 & 25 & 0.118 & 0.118 & 13 & 13 & 0.142 & 0.142 & 0.122 & 0.003000000\\\\\n",
       "\tSAMEA3136634 & control & Feng\\_2015\\_NatComm & CRC & 113 & 113 & 16 & 16 & 0.052 & 0.052 & 16 & 16 & 0.296 & 0.296 & 0.090 & 0.010571429\\\\\n",
       "\tSAMEA3136738 & case    & Feng\\_2015\\_NatComm & CRC & 119 & 119 & 54 & 54 & 0.524 & 0.524 &  5 &  5 & 0.016 & 0.016 & 0.070 & 0.003714286\\\\\n",
       "\tSAMEA3136674 & control & Feng\\_2015\\_NatComm & CRC &  64 &  64 &  6 &  6 & 0.024 & 0.024 & 17 & 17 & 0.548 & 0.548 & 0.112 & 0.026000000\\\\\n",
       "\tSAMEA3136663 & control & Feng\\_2015\\_NatComm & CRC &  80 &  80 & 10 & 10 & 0.058 & 0.058 & 18 & 18 & 0.522 & 0.522 & 0.266 & 0.055200000\\\\\n",
       "\tSAMEA3136747 & case    & Feng\\_2015\\_NatComm & CRC &  62 &  62 &  9 &  9 & 0.068 & 0.068 & 10 & 10 & 0.202 & 0.202 & 0.172 & 0.008000000\\\\\n",
       "\tSAMEA3136676 & control & Feng\\_2015\\_NatComm & CRC & 113 & 113 & 22 & 22 & 0.134 & 0.134 & 18 & 18 & 0.348 & 0.348 & 0.160 & 0.027428571\\\\\n",
       "\tSAMEA3136743 & case    & Feng\\_2015\\_NatComm & CRC & 118 & 118 & 19 & 19 & 0.116 & 0.116 & 13 & 13 & 0.220 & 0.220 & 0.112 & 0.020285714\\\\\n",
       "\tSAMEA3136754 & case    & Feng\\_2015\\_NatComm & CRC & 118 & 118 & 20 & 20 & 0.170 & 0.170 & 18 & 18 & 0.218 & 0.218 & 0.086 & 0.014857143\\\\\n",
       "\tSAMEA3136644 & control & Feng\\_2015\\_NatComm & CRC & 141 & 141 & 32 & 32 & 0.260 & 0.260 & 20 & 20 & 0.194 & 0.194 & 0.070 & 0.008888889\\\\\n",
       "\tSAMEA3136654 & control & Feng\\_2015\\_NatComm & CRC & 128 & 128 & 24 & 24 & 0.148 & 0.148 & 21 & 21 & 0.240 & 0.240 & 0.060 & 0.007250000\\\\\n",
       "\tSAMEA3136673 & control & Feng\\_2015\\_NatComm & CRC &  88 &  88 & 12 & 12 & 0.114 & 0.114 & 17 & 17 & 0.448 & 0.448 & 0.124 & 0.023666667\\\\\n",
       "\tSAMEA3136665 & control & Feng\\_2015\\_NatComm & CRC &  57 &  57 & 15 & 15 & 0.312 & 0.312 & 16 & 16 & 0.160 & 0.160 & 0.256 & 0.011000000\\\\\n",
       "\tSAMEA3136664 & control & Feng\\_2015\\_NatComm & CRC &  88 &  88 &  5 &  5 & 0.026 & 0.026 & 21 & 21 & 0.288 & 0.288 & 0.158 & 0.004333333\\\\\n",
       "\tSAMEA3136765 & case    & Feng\\_2015\\_NatComm & CRC &  83 &  83 & 20 & 20 & 0.072 & 0.072 & 13 & 13 & 0.198 & 0.198 & 0.470 & 0.008666667\\\\\n",
       "\tSAMEA3136656 & control & Feng\\_2015\\_NatComm & CRC &  70 &  70 & 10 & 10 & 0.024 & 0.024 & 13 & 13 & 0.408 & 0.408 & 0.180 & 0.007200000\\\\\n",
       "\tSAMEA3136650 & control & Feng\\_2015\\_NatComm & CRC &  40 &  40 & 17 & 17 & 0.158 & 0.158 &  4 &  4 & 0.666 & 0.666 & 0.540 & 0.004000000\\\\\n",
       "\tSAMEA3136626 & control & Feng\\_2015\\_NatComm & CRC &  48 &  48 &  8 &  8 & 0.046 & 0.046 &  9 &  9 & 0.672 & 0.672 & 0.264 & 0.019000000\\\\\n",
       "\tSAMEA3136724 & case    & Feng\\_2015\\_NatComm & CRC &  70 &  70 &  7 &  7 & 0.022 & 0.022 & 16 & 16 & 0.378 & 0.378 & 0.128 & 0.005200000\\\\\n",
       "\tSAMEA3136668 & control & Feng\\_2015\\_NatComm & CRC &  58 &  58 & 15 & 15 & 0.074 & 0.074 &  7 &  7 & 0.308 & 0.308 & 0.222 & 0.003000000\\\\\n",
       "\tSAMEA3136645 & control & Feng\\_2015\\_NatComm & CRC &  89 &  89 & 15 & 15 & 0.064 & 0.064 & 14 & 14 & 0.210 & 0.210 & 0.264 & 0.010333333\\\\\n",
       "\tSAMEA3136755 & case    & Feng\\_2015\\_NatComm & CRC & 109 & 109 & 30 & 30 & 0.128 & 0.128 & 15 & 15 & 0.300 & 0.300 & 0.152 & 0.002571429\\\\\n",
       "\t⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮ & ⋮\\\\\n",
       "\tWGC076504 & control & Zhu\\_2018\\_Microbiome & BC &  52 &  52 &  7 &  7 & 0.052 & 0.052 & 10 & 10 & 0.420 & 0.420 & 0.176 & 0.010500000\\\\\n",
       "\tWGC076505 & control & Zhu\\_2018\\_Microbiome & BC & 108 & 108 & 13 & 13 & 0.094 & 0.094 & 19 & 19 & 0.254 & 0.254 & 0.174 & 0.006285714\\\\\n",
       "\tWGC076506 & control & Zhu\\_2018\\_Microbiome & BC &  68 &  68 & 12 & 12 & 0.070 & 0.070 & 16 & 16 & 0.540 & 0.540 & 0.228 & 0.003600000\\\\\n",
       "\tWGC076507 & control & Zhu\\_2018\\_Microbiome & BC &  76 &  76 &  5 &  5 & 0.068 & 0.068 & 16 & 16 & 0.124 & 0.124 & 0.256 & 0.006000000\\\\\n",
       "\tWGC076508 & control & Zhu\\_2018\\_Microbiome & BC & 109 & 109 & 11 & 11 & 0.024 & 0.024 & 20 & 20 & 0.192 & 0.192 & 0.254 & 0.007142857\\\\\n",
       "\tWGC076509 & control & Zhu\\_2018\\_Microbiome & BC &  86 &  86 & 10 & 10 & 0.062 & 0.062 & 21 & 21 & 0.194 & 0.194 & 0.250 & 0.004333333\\\\\n",
       "\tWGC076510 & control & Zhu\\_2018\\_Microbiome & BC &  69 &  69 &  4 &  4 & 0.014 & 0.014 & 18 & 18 & 0.528 & 0.528 & 0.194 & 0.004400000\\\\\n",
       "\tWGC076511 & control & Zhu\\_2018\\_Microbiome & BC & 111 & 111 &  9 &  9 & 0.018 & 0.018 & 20 & 20 & 0.418 & 0.418 & 0.092 & 0.003714286\\\\\n",
       "\tWGC076513 & control & Zhu\\_2018\\_Microbiome & BC &  39 &  39 &  3 &  3 & 0.006 & 0.006 &  9 &  9 & 0.776 & 0.776 & 0.424 & 0.026666667\\\\\n",
       "\tWGC076515 & control & Zhu\\_2018\\_Microbiome & BC &  67 &  67 &  3 &  3 & 0.010 & 0.010 & 17 & 17 & 0.372 & 0.372 & 0.222 & 0.010400000\\\\\n",
       "\tWGC076516 & control & Zhu\\_2018\\_Microbiome & BC &  81 &  81 &  6 &  6 & 0.034 & 0.034 & 16 & 16 & 0.340 & 0.340 & 0.118 & 0.006000000\\\\\n",
       "\tWGC076517 & control & Zhu\\_2018\\_Microbiome & BC &  59 &  59 &  4 &  4 & 0.060 & 0.060 & 13 & 13 & 0.466 & 0.466 & 0.212 & 0.007000000\\\\\n",
       "\tWGC076518 & control & Zhu\\_2018\\_Microbiome & BC &  30 &  30 &  2 &  2 & 0.008 & 0.008 & 14 & 14 & 0.614 & 0.614 & 0.484 & 0.006000000\\\\\n",
       "\tWGC076520 & control & Zhu\\_2018\\_Microbiome & BC &  88 &  88 & 18 & 18 & 0.064 & 0.064 & 17 & 17 & 0.410 & 0.410 & 0.184 & 0.003000000\\\\\n",
       "\tWGC076521 & control & Zhu\\_2018\\_Microbiome & BC &  55 &  55 & 15 & 15 & 0.058 & 0.058 &  6 &  6 & 0.466 & 0.466 & 0.246 & 0.002500000\\\\\n",
       "\tWGC076522 & control & Zhu\\_2018\\_Microbiome & BC &  41 &  41 &  4 &  4 & 0.570 & 0.570 & 15 & 15 & 0.152 & 0.152 & 0.306 & 0.013500000\\\\\n",
       "\tWGC076523 & control & Zhu\\_2018\\_Microbiome & BC &  60 &  60 &  5 &  5 & 0.026 & 0.026 & 14 & 14 & 0.622 & 0.622 & 0.400 & 0.006500000\\\\\n",
       "\tWGC076524 & control & Zhu\\_2018\\_Microbiome & BC &  73 &  73 &  4 &  4 & 0.022 & 0.022 & 21 & 21 & 0.362 & 0.362 & 0.200 & 0.004000000\\\\\n",
       "\tWGC076525 & control & Zhu\\_2018\\_Microbiome & BC &  41 &  41 &  4 &  4 & 0.256 & 0.256 &  9 &  9 & 0.078 & 0.078 & 0.460 & 0.063500000\\\\\n",
       "\tWGC076526 & control & Zhu\\_2018\\_Microbiome & BC &  51 &  51 &  4 &  4 & 0.008 & 0.008 & 12 & 12 & 0.672 & 0.672 & 0.218 & 0.002000000\\\\\n",
       "\tWGC076527 & control & Zhu\\_2018\\_Microbiome & BC &  65 &  65 &  4 &  4 & 0.008 & 0.008 & 16 & 16 & 0.312 & 0.312 & 0.184 & 0.003600000\\\\\n",
       "\tWGC076528 & control & Zhu\\_2018\\_Microbiome & BC & 102 & 102 & 10 & 10 & 0.068 & 0.068 & 19 & 19 & 0.366 & 0.366 & 0.138 & 0.007428571\\\\\n",
       "\tWGC076529 & control & Zhu\\_2018\\_Microbiome & BC &  27 &  27 &  9 &  9 & 0.078 & 0.078 &  4 &  4 & 0.646 & 0.646 & 0.348 & 0.013333333\\\\\n",
       "\tWGC076531 & control & Zhu\\_2018\\_Microbiome & BC &  72 &  72 &  6 &  6 & 0.136 & 0.136 & 16 & 16 & 0.394 & 0.394 & 0.158 & 0.006400000\\\\\n",
       "\tWGC076532 & control & Zhu\\_2018\\_Microbiome & BC &  36 &  36 &  7 &  7 & 0.060 & 0.060 & 10 & 10 & 0.184 & 0.184 & 0.502 & 0.014666667\\\\\n",
       "\tWGC076533 & control & Zhu\\_2018\\_Microbiome & BC &  51 &  51 &  6 &  6 & 0.014 & 0.014 & 15 & 15 & 0.468 & 0.468 & 0.332 & 0.005500000\\\\\n",
       "\tWGC076534 & control & Zhu\\_2018\\_Microbiome & BC &  36 &  36 &  3 &  3 & 0.012 & 0.012 & 10 & 10 & 0.306 & 0.306 & 0.452 & 0.003333333\\\\\n",
       "\tWGC076537 & control & Zhu\\_2018\\_Microbiome & BC &  53 &  53 &  9 &  9 & 0.036 & 0.036 & 11 & 11 & 0.708 & 0.708 & 0.510 & 0.004000000\\\\\n",
       "\tWGC076538 & control & Zhu\\_2018\\_Microbiome & BC &  68 &  68 &  7 &  7 & 0.014 & 0.014 & 16 & 16 & 0.544 & 0.544 & 0.290 & 0.005200000\\\\\n",
       "\tWGC076539 & control & Zhu\\_2018\\_Microbiome & BC &  79 &  79 &  3 &  3 & 0.006 & 0.006 & 19 & 19 & 0.586 & 0.586 & 0.234 & 0.007200000\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6497 × 15\n",
       "\n",
       "| <!--/--> | caseControls &lt;fct&gt; | dataset &lt;chr&gt; | condition &lt;chr&gt; | richness &lt;dbl&gt; | richnessWithCutoff &lt;dbl&gt; | richnessAbsentInHC &lt;dbl&gt; | richnessWithCutoffAbsentInHCWithCutoff &lt;dbl&gt; | cumAbundanceAbsentInHC &lt;dbl&gt; | cumAbundanceAbsentInHCWithCutoff &lt;dbl&gt; | richnessPresentInHC &lt;dbl&gt; | richnessWithCutoffPresentInHCWithCutoff &lt;dbl&gt; | cumAbundancePresentInHC &lt;dbl&gt; | cumAbundancePresentInHCWithCutoff &lt;dbl&gt; | topTaxonRelAb &lt;dbl&gt; | topTaxaRelAb &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SAMEA3136748 | case    | Feng_2015_NatComm | CRC |  87 |  87 |  8 |  8 | 0.022 | 0.022 | 18 | 18 | 0.350 | 0.350 | 0.064 | 0.013666667 |\n",
       "| SAMEA3136648 | control | Feng_2015_NatComm | CRC | 100 | 100 | 27 | 27 | 0.260 | 0.260 | 13 | 13 | 0.190 | 0.190 | 0.110 | 0.017000000 |\n",
       "| SAMEA3136638 | control | Feng_2015_NatComm | CRC |  99 |  99 | 14 | 14 | 0.040 | 0.040 | 18 | 18 | 0.382 | 0.382 | 0.078 | 0.003666667 |\n",
       "| SAMEA3136670 | control | Feng_2015_NatComm | CRC |  93 |  93 | 15 | 15 | 0.058 | 0.058 | 14 | 14 | 0.272 | 0.272 | 0.086 | 0.004666667 |\n",
       "| SAMEA3136657 | control | Feng_2015_NatComm | CRC |  49 |  49 | 16 | 16 | 0.162 | 0.162 |  7 |  7 | 0.276 | 0.276 | 0.150 | 0.006500000 |\n",
       "| SAMEA3136660 | control | Feng_2015_NatComm | CRC |  75 |  75 | 17 | 17 | 0.050 | 0.050 | 10 | 10 | 0.340 | 0.340 | 0.208 | 0.008800000 |\n",
       "| SAMEA3136769 | case    | Feng_2015_NatComm | CRC |  49 |  49 | 10 | 10 | 0.064 | 0.064 |  6 |  6 | 0.156 | 0.156 | 0.204 | 0.005000000 |\n",
       "| SAMEA3136630 | control | Feng_2015_NatComm | CRC | 107 | 107 | 27 | 27 | 0.164 | 0.164 | 14 | 14 | 0.360 | 0.360 | 0.068 | 0.003714286 |\n",
       "| SAMEA3136751 | case    | Feng_2015_NatComm | CRC | 124 | 124 | 25 | 25 | 0.118 | 0.118 | 13 | 13 | 0.142 | 0.142 | 0.122 | 0.003000000 |\n",
       "| SAMEA3136634 | control | Feng_2015_NatComm | CRC | 113 | 113 | 16 | 16 | 0.052 | 0.052 | 16 | 16 | 0.296 | 0.296 | 0.090 | 0.010571429 |\n",
       "| SAMEA3136738 | case    | Feng_2015_NatComm | CRC | 119 | 119 | 54 | 54 | 0.524 | 0.524 |  5 |  5 | 0.016 | 0.016 | 0.070 | 0.003714286 |\n",
       "| SAMEA3136674 | control | Feng_2015_NatComm | CRC |  64 |  64 |  6 |  6 | 0.024 | 0.024 | 17 | 17 | 0.548 | 0.548 | 0.112 | 0.026000000 |\n",
       "| SAMEA3136663 | control | Feng_2015_NatComm | CRC |  80 |  80 | 10 | 10 | 0.058 | 0.058 | 18 | 18 | 0.522 | 0.522 | 0.266 | 0.055200000 |\n",
       "| SAMEA3136747 | case    | Feng_2015_NatComm | CRC |  62 |  62 |  9 |  9 | 0.068 | 0.068 | 10 | 10 | 0.202 | 0.202 | 0.172 | 0.008000000 |\n",
       "| SAMEA3136676 | control | Feng_2015_NatComm | CRC | 113 | 113 | 22 | 22 | 0.134 | 0.134 | 18 | 18 | 0.348 | 0.348 | 0.160 | 0.027428571 |\n",
       "| SAMEA3136743 | case    | Feng_2015_NatComm | CRC | 118 | 118 | 19 | 19 | 0.116 | 0.116 | 13 | 13 | 0.220 | 0.220 | 0.112 | 0.020285714 |\n",
       "| SAMEA3136754 | case    | Feng_2015_NatComm | CRC | 118 | 118 | 20 | 20 | 0.170 | 0.170 | 18 | 18 | 0.218 | 0.218 | 0.086 | 0.014857143 |\n",
       "| SAMEA3136644 | control | Feng_2015_NatComm | CRC | 141 | 141 | 32 | 32 | 0.260 | 0.260 | 20 | 20 | 0.194 | 0.194 | 0.070 | 0.008888889 |\n",
       "| SAMEA3136654 | control | Feng_2015_NatComm | CRC | 128 | 128 | 24 | 24 | 0.148 | 0.148 | 21 | 21 | 0.240 | 0.240 | 0.060 | 0.007250000 |\n",
       "| SAMEA3136673 | control | Feng_2015_NatComm | CRC |  88 |  88 | 12 | 12 | 0.114 | 0.114 | 17 | 17 | 0.448 | 0.448 | 0.124 | 0.023666667 |\n",
       "| SAMEA3136665 | control | Feng_2015_NatComm | CRC |  57 |  57 | 15 | 15 | 0.312 | 0.312 | 16 | 16 | 0.160 | 0.160 | 0.256 | 0.011000000 |\n",
       "| SAMEA3136664 | control | Feng_2015_NatComm | CRC |  88 |  88 |  5 |  5 | 0.026 | 0.026 | 21 | 21 | 0.288 | 0.288 | 0.158 | 0.004333333 |\n",
       "| SAMEA3136765 | case    | Feng_2015_NatComm | CRC |  83 |  83 | 20 | 20 | 0.072 | 0.072 | 13 | 13 | 0.198 | 0.198 | 0.470 | 0.008666667 |\n",
       "| SAMEA3136656 | control | Feng_2015_NatComm | CRC |  70 |  70 | 10 | 10 | 0.024 | 0.024 | 13 | 13 | 0.408 | 0.408 | 0.180 | 0.007200000 |\n",
       "| SAMEA3136650 | control | Feng_2015_NatComm | CRC |  40 |  40 | 17 | 17 | 0.158 | 0.158 |  4 |  4 | 0.666 | 0.666 | 0.540 | 0.004000000 |\n",
       "| SAMEA3136626 | control | Feng_2015_NatComm | CRC |  48 |  48 |  8 |  8 | 0.046 | 0.046 |  9 |  9 | 0.672 | 0.672 | 0.264 | 0.019000000 |\n",
       "| SAMEA3136724 | case    | Feng_2015_NatComm | CRC |  70 |  70 |  7 |  7 | 0.022 | 0.022 | 16 | 16 | 0.378 | 0.378 | 0.128 | 0.005200000 |\n",
       "| SAMEA3136668 | control | Feng_2015_NatComm | CRC |  58 |  58 | 15 | 15 | 0.074 | 0.074 |  7 |  7 | 0.308 | 0.308 | 0.222 | 0.003000000 |\n",
       "| SAMEA3136645 | control | Feng_2015_NatComm | CRC |  89 |  89 | 15 | 15 | 0.064 | 0.064 | 14 | 14 | 0.210 | 0.210 | 0.264 | 0.010333333 |\n",
       "| SAMEA3136755 | case    | Feng_2015_NatComm | CRC | 109 | 109 | 30 | 30 | 0.128 | 0.128 | 15 | 15 | 0.300 | 0.300 | 0.152 | 0.002571429 |\n",
       "| ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ | ⋮ |\n",
       "| WGC076504 | control | Zhu_2018_Microbiome | BC |  52 |  52 |  7 |  7 | 0.052 | 0.052 | 10 | 10 | 0.420 | 0.420 | 0.176 | 0.010500000 |\n",
       "| WGC076505 | control | Zhu_2018_Microbiome | BC | 108 | 108 | 13 | 13 | 0.094 | 0.094 | 19 | 19 | 0.254 | 0.254 | 0.174 | 0.006285714 |\n",
       "| WGC076506 | control | Zhu_2018_Microbiome | BC |  68 |  68 | 12 | 12 | 0.070 | 0.070 | 16 | 16 | 0.540 | 0.540 | 0.228 | 0.003600000 |\n",
       "| WGC076507 | control | Zhu_2018_Microbiome | BC |  76 |  76 |  5 |  5 | 0.068 | 0.068 | 16 | 16 | 0.124 | 0.124 | 0.256 | 0.006000000 |\n",
       "| WGC076508 | control | Zhu_2018_Microbiome | BC | 109 | 109 | 11 | 11 | 0.024 | 0.024 | 20 | 20 | 0.192 | 0.192 | 0.254 | 0.007142857 |\n",
       "| WGC076509 | control | Zhu_2018_Microbiome | BC |  86 |  86 | 10 | 10 | 0.062 | 0.062 | 21 | 21 | 0.194 | 0.194 | 0.250 | 0.004333333 |\n",
       "| WGC076510 | control | Zhu_2018_Microbiome | BC |  69 |  69 |  4 |  4 | 0.014 | 0.014 | 18 | 18 | 0.528 | 0.528 | 0.194 | 0.004400000 |\n",
       "| WGC076511 | control | Zhu_2018_Microbiome | BC | 111 | 111 |  9 |  9 | 0.018 | 0.018 | 20 | 20 | 0.418 | 0.418 | 0.092 | 0.003714286 |\n",
       "| WGC076513 | control | Zhu_2018_Microbiome | BC |  39 |  39 |  3 |  3 | 0.006 | 0.006 |  9 |  9 | 0.776 | 0.776 | 0.424 | 0.026666667 |\n",
       "| WGC076515 | control | Zhu_2018_Microbiome | BC |  67 |  67 |  3 |  3 | 0.010 | 0.010 | 17 | 17 | 0.372 | 0.372 | 0.222 | 0.010400000 |\n",
       "| WGC076516 | control | Zhu_2018_Microbiome | BC |  81 |  81 |  6 |  6 | 0.034 | 0.034 | 16 | 16 | 0.340 | 0.340 | 0.118 | 0.006000000 |\n",
       "| WGC076517 | control | Zhu_2018_Microbiome | BC |  59 |  59 |  4 |  4 | 0.060 | 0.060 | 13 | 13 | 0.466 | 0.466 | 0.212 | 0.007000000 |\n",
       "| WGC076518 | control | Zhu_2018_Microbiome | BC |  30 |  30 |  2 |  2 | 0.008 | 0.008 | 14 | 14 | 0.614 | 0.614 | 0.484 | 0.006000000 |\n",
       "| WGC076520 | control | Zhu_2018_Microbiome | BC |  88 |  88 | 18 | 18 | 0.064 | 0.064 | 17 | 17 | 0.410 | 0.410 | 0.184 | 0.003000000 |\n",
       "| WGC076521 | control | Zhu_2018_Microbiome | BC |  55 |  55 | 15 | 15 | 0.058 | 0.058 |  6 |  6 | 0.466 | 0.466 | 0.246 | 0.002500000 |\n",
       "| WGC076522 | control | Zhu_2018_Microbiome | BC |  41 |  41 |  4 |  4 | 0.570 | 0.570 | 15 | 15 | 0.152 | 0.152 | 0.306 | 0.013500000 |\n",
       "| WGC076523 | control | Zhu_2018_Microbiome | BC |  60 |  60 |  5 |  5 | 0.026 | 0.026 | 14 | 14 | 0.622 | 0.622 | 0.400 | 0.006500000 |\n",
       "| WGC076524 | control | Zhu_2018_Microbiome | BC |  73 |  73 |  4 |  4 | 0.022 | 0.022 | 21 | 21 | 0.362 | 0.362 | 0.200 | 0.004000000 |\n",
       "| WGC076525 | control | Zhu_2018_Microbiome | BC |  41 |  41 |  4 |  4 | 0.256 | 0.256 |  9 |  9 | 0.078 | 0.078 | 0.460 | 0.063500000 |\n",
       "| WGC076526 | control | Zhu_2018_Microbiome | BC |  51 |  51 |  4 |  4 | 0.008 | 0.008 | 12 | 12 | 0.672 | 0.672 | 0.218 | 0.002000000 |\n",
       "| WGC076527 | control | Zhu_2018_Microbiome | BC |  65 |  65 |  4 |  4 | 0.008 | 0.008 | 16 | 16 | 0.312 | 0.312 | 0.184 | 0.003600000 |\n",
       "| WGC076528 | control | Zhu_2018_Microbiome | BC | 102 | 102 | 10 | 10 | 0.068 | 0.068 | 19 | 19 | 0.366 | 0.366 | 0.138 | 0.007428571 |\n",
       "| WGC076529 | control | Zhu_2018_Microbiome | BC |  27 |  27 |  9 |  9 | 0.078 | 0.078 |  4 |  4 | 0.646 | 0.646 | 0.348 | 0.013333333 |\n",
       "| WGC076531 | control | Zhu_2018_Microbiome | BC |  72 |  72 |  6 |  6 | 0.136 | 0.136 | 16 | 16 | 0.394 | 0.394 | 0.158 | 0.006400000 |\n",
       "| WGC076532 | control | Zhu_2018_Microbiome | BC |  36 |  36 |  7 |  7 | 0.060 | 0.060 | 10 | 10 | 0.184 | 0.184 | 0.502 | 0.014666667 |\n",
       "| WGC076533 | control | Zhu_2018_Microbiome | BC |  51 |  51 |  6 |  6 | 0.014 | 0.014 | 15 | 15 | 0.468 | 0.468 | 0.332 | 0.005500000 |\n",
       "| WGC076534 | control | Zhu_2018_Microbiome | BC |  36 |  36 |  3 |  3 | 0.012 | 0.012 | 10 | 10 | 0.306 | 0.306 | 0.452 | 0.003333333 |\n",
       "| WGC076537 | control | Zhu_2018_Microbiome | BC |  53 |  53 |  9 |  9 | 0.036 | 0.036 | 11 | 11 | 0.708 | 0.708 | 0.510 | 0.004000000 |\n",
       "| WGC076538 | control | Zhu_2018_Microbiome | BC |  68 |  68 |  7 |  7 | 0.014 | 0.014 | 16 | 16 | 0.544 | 0.544 | 0.290 | 0.005200000 |\n",
       "| WGC076539 | control | Zhu_2018_Microbiome | BC |  79 |  79 |  3 |  3 | 0.006 | 0.006 | 19 | 19 | 0.586 | 0.586 | 0.234 | 0.007200000 |\n",
       "\n"
      ],
      "text/plain": [
       "             caseControls dataset             condition richness\n",
       "SAMEA3136748 case         Feng_2015_NatComm   CRC        87     \n",
       "SAMEA3136648 control      Feng_2015_NatComm   CRC       100     \n",
       "SAMEA3136638 control      Feng_2015_NatComm   CRC        99     \n",
       "SAMEA3136670 control      Feng_2015_NatComm   CRC        93     \n",
       "SAMEA3136657 control      Feng_2015_NatComm   CRC        49     \n",
       "SAMEA3136660 control      Feng_2015_NatComm   CRC        75     \n",
       "SAMEA3136769 case         Feng_2015_NatComm   CRC        49     \n",
       "SAMEA3136630 control      Feng_2015_NatComm   CRC       107     \n",
       "SAMEA3136751 case         Feng_2015_NatComm   CRC       124     \n",
       "SAMEA3136634 control      Feng_2015_NatComm   CRC       113     \n",
       "SAMEA3136738 case         Feng_2015_NatComm   CRC       119     \n",
       "SAMEA3136674 control      Feng_2015_NatComm   CRC        64     \n",
       "SAMEA3136663 control      Feng_2015_NatComm   CRC        80     \n",
       "SAMEA3136747 case         Feng_2015_NatComm   CRC        62     \n",
       "SAMEA3136676 control      Feng_2015_NatComm   CRC       113     \n",
       "SAMEA3136743 case         Feng_2015_NatComm   CRC       118     \n",
       "SAMEA3136754 case         Feng_2015_NatComm   CRC       118     \n",
       "SAMEA3136644 control      Feng_2015_NatComm   CRC       141     \n",
       "SAMEA3136654 control      Feng_2015_NatComm   CRC       128     \n",
       "SAMEA3136673 control      Feng_2015_NatComm   CRC        88     \n",
       "SAMEA3136665 control      Feng_2015_NatComm   CRC        57     \n",
       "SAMEA3136664 control      Feng_2015_NatComm   CRC        88     \n",
       "SAMEA3136765 case         Feng_2015_NatComm   CRC        83     \n",
       "SAMEA3136656 control      Feng_2015_NatComm   CRC        70     \n",
       "SAMEA3136650 control      Feng_2015_NatComm   CRC        40     \n",
       "SAMEA3136626 control      Feng_2015_NatComm   CRC        48     \n",
       "SAMEA3136724 case         Feng_2015_NatComm   CRC        70     \n",
       "SAMEA3136668 control      Feng_2015_NatComm   CRC        58     \n",
       "SAMEA3136645 control      Feng_2015_NatComm   CRC        89     \n",
       "SAMEA3136755 case         Feng_2015_NatComm   CRC       109     \n",
       "⋮            ⋮            ⋮                   ⋮         ⋮       \n",
       "WGC076504    control      Zhu_2018_Microbiome BC         52     \n",
       "WGC076505    control      Zhu_2018_Microbiome BC        108     \n",
       "WGC076506    control      Zhu_2018_Microbiome BC         68     \n",
       "WGC076507    control      Zhu_2018_Microbiome BC         76     \n",
       "WGC076508    control      Zhu_2018_Microbiome BC        109     \n",
       "WGC076509    control      Zhu_2018_Microbiome BC         86     \n",
       "WGC076510    control      Zhu_2018_Microbiome BC         69     \n",
       "WGC076511    control      Zhu_2018_Microbiome BC        111     \n",
       "WGC076513    control      Zhu_2018_Microbiome BC         39     \n",
       "WGC076515    control      Zhu_2018_Microbiome BC         67     \n",
       "WGC076516    control      Zhu_2018_Microbiome BC         81     \n",
       "WGC076517    control      Zhu_2018_Microbiome BC         59     \n",
       "WGC076518    control      Zhu_2018_Microbiome BC         30     \n",
       "WGC076520    control      Zhu_2018_Microbiome BC         88     \n",
       "WGC076521    control      Zhu_2018_Microbiome BC         55     \n",
       "WGC076522    control      Zhu_2018_Microbiome BC         41     \n",
       "WGC076523    control      Zhu_2018_Microbiome BC         60     \n",
       "WGC076524    control      Zhu_2018_Microbiome BC         73     \n",
       "WGC076525    control      Zhu_2018_Microbiome BC         41     \n",
       "WGC076526    control      Zhu_2018_Microbiome BC         51     \n",
       "WGC076527    control      Zhu_2018_Microbiome BC         65     \n",
       "WGC076528    control      Zhu_2018_Microbiome BC        102     \n",
       "WGC076529    control      Zhu_2018_Microbiome BC         27     \n",
       "WGC076531    control      Zhu_2018_Microbiome BC         72     \n",
       "WGC076532    control      Zhu_2018_Microbiome BC         36     \n",
       "WGC076533    control      Zhu_2018_Microbiome BC         51     \n",
       "WGC076534    control      Zhu_2018_Microbiome BC         36     \n",
       "WGC076537    control      Zhu_2018_Microbiome BC         53     \n",
       "WGC076538    control      Zhu_2018_Microbiome BC         68     \n",
       "WGC076539    control      Zhu_2018_Microbiome BC         79     \n",
       "             richnessWithCutoff richnessAbsentInHC\n",
       "SAMEA3136748  87                 8                \n",
       "SAMEA3136648 100                27                \n",
       "SAMEA3136638  99                14                \n",
       "SAMEA3136670  93                15                \n",
       "SAMEA3136657  49                16                \n",
       "SAMEA3136660  75                17                \n",
       "SAMEA3136769  49                10                \n",
       "SAMEA3136630 107                27                \n",
       "SAMEA3136751 124                25                \n",
       "SAMEA3136634 113                16                \n",
       "SAMEA3136738 119                54                \n",
       "SAMEA3136674  64                 6                \n",
       "SAMEA3136663  80                10                \n",
       "SAMEA3136747  62                 9                \n",
       "SAMEA3136676 113                22                \n",
       "SAMEA3136743 118                19                \n",
       "SAMEA3136754 118                20                \n",
       "SAMEA3136644 141                32                \n",
       "SAMEA3136654 128                24                \n",
       "SAMEA3136673  88                12                \n",
       "SAMEA3136665  57                15                \n",
       "SAMEA3136664  88                 5                \n",
       "SAMEA3136765  83                20                \n",
       "SAMEA3136656  70                10                \n",
       "SAMEA3136650  40                17                \n",
       "SAMEA3136626  48                 8                \n",
       "SAMEA3136724  70                 7                \n",
       "SAMEA3136668  58                15                \n",
       "SAMEA3136645  89                15                \n",
       "SAMEA3136755 109                30                \n",
       "⋮            ⋮                  ⋮                 \n",
       "WGC076504     52                 7                \n",
       "WGC076505    108                13                \n",
       "WGC076506     68                12                \n",
       "WGC076507     76                 5                \n",
       "WGC076508    109                11                \n",
       "WGC076509     86                10                \n",
       "WGC076510     69                 4                \n",
       "WGC076511    111                 9                \n",
       "WGC076513     39                 3                \n",
       "WGC076515     67                 3                \n",
       "WGC076516     81                 6                \n",
       "WGC076517     59                 4                \n",
       "WGC076518     30                 2                \n",
       "WGC076520     88                18                \n",
       "WGC076521     55                15                \n",
       "WGC076522     41                 4                \n",
       "WGC076523     60                 5                \n",
       "WGC076524     73                 4                \n",
       "WGC076525     41                 4                \n",
       "WGC076526     51                 4                \n",
       "WGC076527     65                 4                \n",
       "WGC076528    102                10                \n",
       "WGC076529     27                 9                \n",
       "WGC076531     72                 6                \n",
       "WGC076532     36                 7                \n",
       "WGC076533     51                 6                \n",
       "WGC076534     36                 3                \n",
       "WGC076537     53                 9                \n",
       "WGC076538     68                 7                \n",
       "WGC076539     79                 3                \n",
       "             richnessWithCutoffAbsentInHCWithCutoff cumAbundanceAbsentInHC\n",
       "SAMEA3136748  8                                     0.022                 \n",
       "SAMEA3136648 27                                     0.260                 \n",
       "SAMEA3136638 14                                     0.040                 \n",
       "SAMEA3136670 15                                     0.058                 \n",
       "SAMEA3136657 16                                     0.162                 \n",
       "SAMEA3136660 17                                     0.050                 \n",
       "SAMEA3136769 10                                     0.064                 \n",
       "SAMEA3136630 27                                     0.164                 \n",
       "SAMEA3136751 25                                     0.118                 \n",
       "SAMEA3136634 16                                     0.052                 \n",
       "SAMEA3136738 54                                     0.524                 \n",
       "SAMEA3136674  6                                     0.024                 \n",
       "SAMEA3136663 10                                     0.058                 \n",
       "SAMEA3136747  9                                     0.068                 \n",
       "SAMEA3136676 22                                     0.134                 \n",
       "SAMEA3136743 19                                     0.116                 \n",
       "SAMEA3136754 20                                     0.170                 \n",
       "SAMEA3136644 32                                     0.260                 \n",
       "SAMEA3136654 24                                     0.148                 \n",
       "SAMEA3136673 12                                     0.114                 \n",
       "SAMEA3136665 15                                     0.312                 \n",
       "SAMEA3136664  5                                     0.026                 \n",
       "SAMEA3136765 20                                     0.072                 \n",
       "SAMEA3136656 10                                     0.024                 \n",
       "SAMEA3136650 17                                     0.158                 \n",
       "SAMEA3136626  8                                     0.046                 \n",
       "SAMEA3136724  7                                     0.022                 \n",
       "SAMEA3136668 15                                     0.074                 \n",
       "SAMEA3136645 15                                     0.064                 \n",
       "SAMEA3136755 30                                     0.128                 \n",
       "⋮            ⋮                                      ⋮                     \n",
       "WGC076504     7                                     0.052                 \n",
       "WGC076505    13                                     0.094                 \n",
       "WGC076506    12                                     0.070                 \n",
       "WGC076507     5                                     0.068                 \n",
       "WGC076508    11                                     0.024                 \n",
       "WGC076509    10                                     0.062                 \n",
       "WGC076510     4                                     0.014                 \n",
       "WGC076511     9                                     0.018                 \n",
       "WGC076513     3                                     0.006                 \n",
       "WGC076515     3                                     0.010                 \n",
       "WGC076516     6                                     0.034                 \n",
       "WGC076517     4                                     0.060                 \n",
       "WGC076518     2                                     0.008                 \n",
       "WGC076520    18                                     0.064                 \n",
       "WGC076521    15                                     0.058                 \n",
       "WGC076522     4                                     0.570                 \n",
       "WGC076523     5                                     0.026                 \n",
       "WGC076524     4                                     0.022                 \n",
       "WGC076525     4                                     0.256                 \n",
       "WGC076526     4                                     0.008                 \n",
       "WGC076527     4                                     0.008                 \n",
       "WGC076528    10                                     0.068                 \n",
       "WGC076529     9                                     0.078                 \n",
       "WGC076531     6                                     0.136                 \n",
       "WGC076532     7                                     0.060                 \n",
       "WGC076533     6                                     0.014                 \n",
       "WGC076534     3                                     0.012                 \n",
       "WGC076537     9                                     0.036                 \n",
       "WGC076538     7                                     0.014                 \n",
       "WGC076539     3                                     0.006                 \n",
       "             cumAbundanceAbsentInHCWithCutoff richnessPresentInHC\n",
       "SAMEA3136748 0.022                            18                 \n",
       "SAMEA3136648 0.260                            13                 \n",
       "SAMEA3136638 0.040                            18                 \n",
       "SAMEA3136670 0.058                            14                 \n",
       "SAMEA3136657 0.162                             7                 \n",
       "SAMEA3136660 0.050                            10                 \n",
       "SAMEA3136769 0.064                             6                 \n",
       "SAMEA3136630 0.164                            14                 \n",
       "SAMEA3136751 0.118                            13                 \n",
       "SAMEA3136634 0.052                            16                 \n",
       "SAMEA3136738 0.524                             5                 \n",
       "SAMEA3136674 0.024                            17                 \n",
       "SAMEA3136663 0.058                            18                 \n",
       "SAMEA3136747 0.068                            10                 \n",
       "SAMEA3136676 0.134                            18                 \n",
       "SAMEA3136743 0.116                            13                 \n",
       "SAMEA3136754 0.170                            18                 \n",
       "SAMEA3136644 0.260                            20                 \n",
       "SAMEA3136654 0.148                            21                 \n",
       "SAMEA3136673 0.114                            17                 \n",
       "SAMEA3136665 0.312                            16                 \n",
       "SAMEA3136664 0.026                            21                 \n",
       "SAMEA3136765 0.072                            13                 \n",
       "SAMEA3136656 0.024                            13                 \n",
       "SAMEA3136650 0.158                             4                 \n",
       "SAMEA3136626 0.046                             9                 \n",
       "SAMEA3136724 0.022                            16                 \n",
       "SAMEA3136668 0.074                             7                 \n",
       "SAMEA3136645 0.064                            14                 \n",
       "SAMEA3136755 0.128                            15                 \n",
       "⋮            ⋮                                ⋮                  \n",
       "WGC076504    0.052                            10                 \n",
       "WGC076505    0.094                            19                 \n",
       "WGC076506    0.070                            16                 \n",
       "WGC076507    0.068                            16                 \n",
       "WGC076508    0.024                            20                 \n",
       "WGC076509    0.062                            21                 \n",
       "WGC076510    0.014                            18                 \n",
       "WGC076511    0.018                            20                 \n",
       "WGC076513    0.006                             9                 \n",
       "WGC076515    0.010                            17                 \n",
       "WGC076516    0.034                            16                 \n",
       "WGC076517    0.060                            13                 \n",
       "WGC076518    0.008                            14                 \n",
       "WGC076520    0.064                            17                 \n",
       "WGC076521    0.058                             6                 \n",
       "WGC076522    0.570                            15                 \n",
       "WGC076523    0.026                            14                 \n",
       "WGC076524    0.022                            21                 \n",
       "WGC076525    0.256                             9                 \n",
       "WGC076526    0.008                            12                 \n",
       "WGC076527    0.008                            16                 \n",
       "WGC076528    0.068                            19                 \n",
       "WGC076529    0.078                             4                 \n",
       "WGC076531    0.136                            16                 \n",
       "WGC076532    0.060                            10                 \n",
       "WGC076533    0.014                            15                 \n",
       "WGC076534    0.012                            10                 \n",
       "WGC076537    0.036                            11                 \n",
       "WGC076538    0.014                            16                 \n",
       "WGC076539    0.006                            19                 \n",
       "             richnessWithCutoffPresentInHCWithCutoff cumAbundancePresentInHC\n",
       "SAMEA3136748 18                                      0.350                  \n",
       "SAMEA3136648 13                                      0.190                  \n",
       "SAMEA3136638 18                                      0.382                  \n",
       "SAMEA3136670 14                                      0.272                  \n",
       "SAMEA3136657  7                                      0.276                  \n",
       "SAMEA3136660 10                                      0.340                  \n",
       "SAMEA3136769  6                                      0.156                  \n",
       "SAMEA3136630 14                                      0.360                  \n",
       "SAMEA3136751 13                                      0.142                  \n",
       "SAMEA3136634 16                                      0.296                  \n",
       "SAMEA3136738  5                                      0.016                  \n",
       "SAMEA3136674 17                                      0.548                  \n",
       "SAMEA3136663 18                                      0.522                  \n",
       "SAMEA3136747 10                                      0.202                  \n",
       "SAMEA3136676 18                                      0.348                  \n",
       "SAMEA3136743 13                                      0.220                  \n",
       "SAMEA3136754 18                                      0.218                  \n",
       "SAMEA3136644 20                                      0.194                  \n",
       "SAMEA3136654 21                                      0.240                  \n",
       "SAMEA3136673 17                                      0.448                  \n",
       "SAMEA3136665 16                                      0.160                  \n",
       "SAMEA3136664 21                                      0.288                  \n",
       "SAMEA3136765 13                                      0.198                  \n",
       "SAMEA3136656 13                                      0.408                  \n",
       "SAMEA3136650  4                                      0.666                  \n",
       "SAMEA3136626  9                                      0.672                  \n",
       "SAMEA3136724 16                                      0.378                  \n",
       "SAMEA3136668  7                                      0.308                  \n",
       "SAMEA3136645 14                                      0.210                  \n",
       "SAMEA3136755 15                                      0.300                  \n",
       "⋮            ⋮                                       ⋮                      \n",
       "WGC076504    10                                      0.420                  \n",
       "WGC076505    19                                      0.254                  \n",
       "WGC076506    16                                      0.540                  \n",
       "WGC076507    16                                      0.124                  \n",
       "WGC076508    20                                      0.192                  \n",
       "WGC076509    21                                      0.194                  \n",
       "WGC076510    18                                      0.528                  \n",
       "WGC076511    20                                      0.418                  \n",
       "WGC076513     9                                      0.776                  \n",
       "WGC076515    17                                      0.372                  \n",
       "WGC076516    16                                      0.340                  \n",
       "WGC076517    13                                      0.466                  \n",
       "WGC076518    14                                      0.614                  \n",
       "WGC076520    17                                      0.410                  \n",
       "WGC076521     6                                      0.466                  \n",
       "WGC076522    15                                      0.152                  \n",
       "WGC076523    14                                      0.622                  \n",
       "WGC076524    21                                      0.362                  \n",
       "WGC076525     9                                      0.078                  \n",
       "WGC076526    12                                      0.672                  \n",
       "WGC076527    16                                      0.312                  \n",
       "WGC076528    19                                      0.366                  \n",
       "WGC076529     4                                      0.646                  \n",
       "WGC076531    16                                      0.394                  \n",
       "WGC076532    10                                      0.184                  \n",
       "WGC076533    15                                      0.468                  \n",
       "WGC076534    10                                      0.306                  \n",
       "WGC076537    11                                      0.708                  \n",
       "WGC076538    16                                      0.544                  \n",
       "WGC076539    19                                      0.586                  \n",
       "             cumAbundancePresentInHCWithCutoff topTaxonRelAb topTaxaRelAb\n",
       "SAMEA3136748 0.350                             0.064         0.013666667 \n",
       "SAMEA3136648 0.190                             0.110         0.017000000 \n",
       "SAMEA3136638 0.382                             0.078         0.003666667 \n",
       "SAMEA3136670 0.272                             0.086         0.004666667 \n",
       "SAMEA3136657 0.276                             0.150         0.006500000 \n",
       "SAMEA3136660 0.340                             0.208         0.008800000 \n",
       "SAMEA3136769 0.156                             0.204         0.005000000 \n",
       "SAMEA3136630 0.360                             0.068         0.003714286 \n",
       "SAMEA3136751 0.142                             0.122         0.003000000 \n",
       "SAMEA3136634 0.296                             0.090         0.010571429 \n",
       "SAMEA3136738 0.016                             0.070         0.003714286 \n",
       "SAMEA3136674 0.548                             0.112         0.026000000 \n",
       "SAMEA3136663 0.522                             0.266         0.055200000 \n",
       "SAMEA3136747 0.202                             0.172         0.008000000 \n",
       "SAMEA3136676 0.348                             0.160         0.027428571 \n",
       "SAMEA3136743 0.220                             0.112         0.020285714 \n",
       "SAMEA3136754 0.218                             0.086         0.014857143 \n",
       "SAMEA3136644 0.194                             0.070         0.008888889 \n",
       "SAMEA3136654 0.240                             0.060         0.007250000 \n",
       "SAMEA3136673 0.448                             0.124         0.023666667 \n",
       "SAMEA3136665 0.160                             0.256         0.011000000 \n",
       "SAMEA3136664 0.288                             0.158         0.004333333 \n",
       "SAMEA3136765 0.198                             0.470         0.008666667 \n",
       "SAMEA3136656 0.408                             0.180         0.007200000 \n",
       "SAMEA3136650 0.666                             0.540         0.004000000 \n",
       "SAMEA3136626 0.672                             0.264         0.019000000 \n",
       "SAMEA3136724 0.378                             0.128         0.005200000 \n",
       "SAMEA3136668 0.308                             0.222         0.003000000 \n",
       "SAMEA3136645 0.210                             0.264         0.010333333 \n",
       "SAMEA3136755 0.300                             0.152         0.002571429 \n",
       "⋮            ⋮                                 ⋮             ⋮           \n",
       "WGC076504    0.420                             0.176         0.010500000 \n",
       "WGC076505    0.254                             0.174         0.006285714 \n",
       "WGC076506    0.540                             0.228         0.003600000 \n",
       "WGC076507    0.124                             0.256         0.006000000 \n",
       "WGC076508    0.192                             0.254         0.007142857 \n",
       "WGC076509    0.194                             0.250         0.004333333 \n",
       "WGC076510    0.528                             0.194         0.004400000 \n",
       "WGC076511    0.418                             0.092         0.003714286 \n",
       "WGC076513    0.776                             0.424         0.026666667 \n",
       "WGC076515    0.372                             0.222         0.010400000 \n",
       "WGC076516    0.340                             0.118         0.006000000 \n",
       "WGC076517    0.466                             0.212         0.007000000 \n",
       "WGC076518    0.614                             0.484         0.006000000 \n",
       "WGC076520    0.410                             0.184         0.003000000 \n",
       "WGC076521    0.466                             0.246         0.002500000 \n",
       "WGC076522    0.152                             0.306         0.013500000 \n",
       "WGC076523    0.622                             0.400         0.006500000 \n",
       "WGC076524    0.362                             0.200         0.004000000 \n",
       "WGC076525    0.078                             0.460         0.063500000 \n",
       "WGC076526    0.672                             0.218         0.002000000 \n",
       "WGC076527    0.312                             0.184         0.003600000 \n",
       "WGC076528    0.366                             0.138         0.007428571 \n",
       "WGC076529    0.646                             0.348         0.013333333 \n",
       "WGC076531    0.394                             0.158         0.006400000 \n",
       "WGC076532    0.184                             0.502         0.014666667 \n",
       "WGC076533    0.468                             0.332         0.005500000 \n",
       "WGC076534    0.306                             0.452         0.003333333 \n",
       "WGC076537    0.708                             0.510         0.004000000 \n",
       "WGC076538    0.544                             0.290         0.005200000 \n",
       "WGC076539    0.586                             0.234         0.007200000 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metaDataWGS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e52bf9a-a31e-401b-b363-9502a6874f5b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check correlations between dataMetaAdaptedTruncWideAll and dataMetaAdaptedTruncWideAllRarefied at this point"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "id": "d83af92a",
   "metadata": {},
   "outputs": [],
   "source": [
    "save(metaDataWGS, dataMetaAdaptedTruncWideAll, dataMetaAdaptedTruncWideAllRarefied, file = '/g/scb2/zeller/karcher/dysb_classif/data/siamcat/profiles_merged_with_metadata.rimage')\n",
    "metaDataWGS %>% write_tsv(\"/g/scb2/zeller/karcher/dysb_classif/data/siamcat/metaDataWGS.tsv\")\n",
    "dataMetaAdaptedTruncWideAll %>% write_tsv(\"/g/scb2/zeller/karcher/dysb_classif/data/siamcat/dataMetaAdaptedTruncWideAll.tsv\")\n",
    "dataMetaAdaptedTruncWideAllRarefied %>% write_tsv(\"/g/scb2/zeller/karcher/dysb_classif/data/siamcat/dataMetaAdaptedTruncWideAllRarefied.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "911c9611-89de-4dc1-81ef-a066e6f1747f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
